From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Bradley <daniel@pulumi.com>
Date: Fri, 4 Nov 2022 16:49:08 +0000
Subject: [PATCH 01/30] Add TagsSchemaTrulyComputed definition


diff --git a/internal/tags/tags.go b/internal/tags/tags.go
index 12d3b9013e..bc9e14fc19 100644
--- a/internal/tags/tags.go
+++ b/internal/tags/tags.go
@@ -33,3 +33,11 @@ func TagsSchemaForceNew() *schema.Schema {
 		Elem:     &schema.Schema{Type: schema.TypeString},
 	}
 }
+
+func TagsSchemaTrulyComputed() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeMap,
+		Computed: true,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+	}
+}
