From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Bradley <daniel@pulumi.com>
Date: Fri, 4 Nov 2022 16:49:08 +0000
Subject: [PATCH] Add TagsSchemaTrulyComputed definition


diff --git a/internal/tags/tags.go b/internal/tags/tags.go
index 40c06bb057f..247a01bba3c 100644
--- a/internal/tags/tags.go
+++ b/internal/tags/tags.go
@@ -37,3 +37,11 @@ var TagsSchemaForceNew = sync.OnceValue(func() *schema.Schema {
 		Elem:     &schema.Schema{Type: schema.TypeString},
 	}
 })
+
+func TagsSchemaTrulyComputed() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeMap,
+		Computed: true,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+	}
+}
