From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Bradley <daniel@pulumi.com>
Date: Fri, 4 Nov 2022 17:06:11 +0000
Subject: [PATCH 04/51] De-deprecate bucket_object


diff --git a/internal/service/s3/bucket_object.go b/internal/service/s3/bucket_object.go
index 853f7f1780..5e7fd2bfd3 100644
--- a/internal/service/s3/bucket_object.go
+++ b/internal/service/s3/bucket_object.go
@@ -67,7 +67,7 @@ func resourceBucketObject() *schema.Resource {
 				Computed: true,
 			},
 			"bucket": {
-				Deprecated:   "Use the aws_s3_object resource instead",
+				// FORK: Stack72 removed the deprecation warning as it was misleading for Pulumi users
 				Type:         schema.TypeString,
 				Required:     true,
 				ForceNew:     true,
@@ -124,7 +124,7 @@ func resourceBucketObject() *schema.Resource {
 				Default:  false,
 			},
 			"key": {
-				Deprecated:   "Use the aws_s3_object resource instead",
+				// FORK: Stack72 removed the deprecation warning as it was misleading for Pulumi users
 				Type:         schema.TypeString,
 				Required:     true,
 				ForceNew:     true,
@@ -197,7 +197,9 @@ func resourceBucketObject() *schema.Resource {
 			},
 		},
 
-		DeprecationMessage: `use the aws_s3_object resource instead`,
+		// We are keeping aws_s3_bucket_object alive just like aws_s3_bucket.
+		//
+		// DeprecationMessage: `use the aws_s3_object resource instead`,
 	}
 }
 
