From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Bradley <daniel@pulumi.com>
Date: Fri, 4 Nov 2022 17:06:11 +0000
Subject: [PATCH] De-deprecate bucket_object


diff --git a/internal/service/s3/bucket_object.go b/internal/service/s3/bucket_object.go
index 85729aea89..53794a2d43 100644
--- a/internal/service/s3/bucket_object.go
+++ b/internal/service/s3/bucket_object.go
@@ -66,7 +66,7 @@ func resourceBucketObject() *schema.Resource {
 				Computed: true,
 			},
 			names.AttrBucket: {
-				Deprecated:   "bucket is deprecated. Use the aws_s3_object resource instead.",
+				// FORK: Stack72 removed the deprecation warning as it was misleading for Pulumi users
 				Type:         schema.TypeString,
 				Required:     true,
 				ForceNew:     true,
@@ -123,7 +123,7 @@ func resourceBucketObject() *schema.Resource {
 				Default:  false,
 			},
 			names.AttrKey: {
-				Deprecated:   "key is deprecated. Use the aws_s3_object resource instead.",
+				// FORK: Stack72 removed the deprecation warning as it was misleading for Pulumi users
 				Type:         schema.TypeString,
 				Required:     true,
 				ForceNew:     true,
@@ -189,7 +189,9 @@ func resourceBucketObject() *schema.Resource {
 			},
 		},
 
-		DeprecationMessage: `use the aws_s3_object resource instead`,
+		// We are keeping aws_s3_bucket_object alive just like aws_s3_bucket.
+		//
+		// DeprecationMessage: `use the aws_s3_object resource instead`,
 	}
 }
 
