From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Bradley <daniel@pulumi.com>
Date: Fri, 4 Nov 2022 17:08:23 +0000
Subject: [PATCH 06/30] Remove lakeformation catalog_resource default


diff --git a/internal/service/lakeformation/permissions.go b/internal/service/lakeformation/permissions.go
index 594d25ee73..258163d0a5 100644
--- a/internal/service/lakeformation/permissions.go
+++ b/internal/service/lakeformation/permissions.go
@@ -40,8 +40,8 @@ func ResourcePermissions() *schema.Resource {
 				ValidateFunc: verify.ValidAccountID,
 			},
 			"catalog_resource": {
-				Type:     schema.TypeBool,
-				Default:  false,
+				Type: schema.TypeBool,
+				//Default:  false,
 				ForceNew: true,
 				Optional: true,
 				ExactlyOneOf: []string{
