From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Bradley <daniel@pulumi.com>
Date: Fri, 4 Nov 2022 17:32:49 +0000
Subject: [PATCH] Add EKS cluster certificate_authorities (plural)


diff --git a/internal/service/eks/cluster.go b/internal/service/eks/cluster.go
index c0f792399f6..c2b5a975a98 100644
--- a/internal/service/eks/cluster.go
+++ b/internal/service/eks/cluster.go
@@ -122,9 +122,24 @@ func resourceCluster() *schema.Resource {
 				ForceNew: true,
 				Default:  true,
 			},
+			// FORK: Stack72: Renamed certificate_authority list to be certificate_authorities and map in the Pulumi provider
+			"certificate_authorities": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"data": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			// FORK: Stack72: Added a singular backward compatible value for certificate authorities
 			"certificate_authority": {
 				Type:     schema.TypeList,
 				Computed: true,
+				//MaxItems: 1,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"data": {
@@ -628,6 +643,13 @@ func resourceClusterRead(ctx context.Context, d *schema.ResourceData, meta inter
 	}
 	d.Set(names.AttrARN, cluster.Arn)
 	d.Set("bootstrap_self_managed_addons", d.Get("bootstrap_self_managed_addons"))
+
+	// FORK: Stack72: Renamed certificate_authority list to be certificate_authorities and map in the Pulumi provider
+	if err := d.Set("certificate_authorities", flattenCertificate(cluster.CertificateAuthority)); err != nil {
+		return diag.Errorf("error setting certificate_authorities: %w", err)
+	}
+
+	// FORK: Stack72: Adding a single certificate authority to ensure backwards compatibility
 	if err := d.Set("certificate_authority", flattenCertificate(cluster.CertificateAuthority)); err != nil {
 		return sdkdiag.AppendErrorf(diags, "setting certificate_authority: %s", err)
 	}
