From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: corymhall <43035978+corymhall@users.noreply.github.com>
Date: Wed, 29 May 2024 09:05:36 -0400
Subject: [PATCH] Change default descriptions to "Managed by Pulumi"


diff --git a/internal/service/apigateway/api_key.go b/internal/service/apigateway/api_key.go
index c77a1a31d91..f4d8cb4c016 100644
--- a/internal/service/apigateway/api_key.go
+++ b/internal/service/apigateway/api_key.go
@@ -54,7 +54,7 @@ func resourceAPIKey() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrEnabled: {
 				Type:     schema.TypeBool,
diff --git a/internal/service/appsync/api_key.go b/internal/service/appsync/api_key.go
index 9bbca4265c2..8880372445f 100644
--- a/internal/service/appsync/api_key.go
+++ b/internal/service/appsync/api_key.go
@@ -49,7 +49,7 @@ func resourceAPIKey() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			"expires": {
 				Type:     schema.TypeString,
diff --git a/internal/service/cloudfront/origin_access_control.go b/internal/service/cloudfront/origin_access_control.go
index 08ea92079f6..ad0004ee9e9 100644
--- a/internal/service/cloudfront/origin_access_control.go
+++ b/internal/service/cloudfront/origin_access_control.go
@@ -42,7 +42,7 @@ func resourceOriginAccessControl() *schema.Resource {
 			names.AttrDescription: {
 				Type:         schema.TypeString,
 				Optional:     true,
-				Default:      "Managed by Terraform",
+				Default:      "Managed by Pulumi",
 				ValidateFunc: validation.StringLenBetween(0, 256),
 			},
 			"etag": {
diff --git a/internal/service/docdb/cluster_parameter_group.go b/internal/service/docdb/cluster_parameter_group.go
index d76992a040b..245a84586a2 100644
--- a/internal/service/docdb/cluster_parameter_group.go
+++ b/internal/service/docdb/cluster_parameter_group.go
@@ -49,7 +49,7 @@ func resourceClusterParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/docdb/subnet_group.go b/internal/service/docdb/subnet_group.go
index 96abcbecf98..bfec91c4990 100644
--- a/internal/service/docdb/subnet_group.go
+++ b/internal/service/docdb/subnet_group.go
@@ -45,7 +45,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:          schema.TypeString,
diff --git a/internal/service/ec2/vpc_security_group.go b/internal/service/ec2/vpc_security_group.go
index b335b96d447..dda4ce39c19 100644
--- a/internal/service/ec2/vpc_security_group.go
+++ b/internal/service/ec2/vpc_security_group.go
@@ -67,7 +67,7 @@ func resourceSecurityGroup() *schema.Resource {
 				Type:         schema.TypeString,
 				Optional:     true,
 				ForceNew:     true,
-				Default:      "Managed by Terraform",
+				Default:      "Managed by Pulumi",
 				ValidateFunc: validation.StringLenBetween(0, 255),
 			},
 			"egress":  securityGroupRuleSetNestedBlock,
diff --git a/internal/service/elasticache/parameter_group.go b/internal/service/elasticache/parameter_group.go
index 4e6a87148dc..ac2e529f977 100644
--- a/internal/service/elasticache/parameter_group.go
+++ b/internal/service/elasticache/parameter_group.go
@@ -49,7 +49,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/elasticache/subnet_group.go b/internal/service/elasticache/subnet_group.go
index e733ba52e0f..bea550d91de 100644
--- a/internal/service/elasticache/subnet_group.go
+++ b/internal/service/elasticache/subnet_group.go
@@ -46,7 +46,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:     schema.TypeString,
diff --git a/internal/service/mediapackage/channel.go b/internal/service/mediapackage/channel.go
index cfc5a0b32e2..060b7a72b6a 100644
--- a/internal/service/mediapackage/channel.go
+++ b/internal/service/mediapackage/channel.go
@@ -50,7 +50,7 @@ func ResourceChannel() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			"hls_ingest": {
 				Type:     schema.TypeList,
diff --git a/internal/service/memorydb/cluster.go b/internal/service/memorydb/cluster.go
index cbb54366b05..77c5ef1c309 100644
--- a/internal/service/memorydb/cluster.go
+++ b/internal/service/memorydb/cluster.go
@@ -77,7 +77,7 @@ func resourceCluster() *schema.Resource {
 				names.AttrDescription: {
 					Type:     schema.TypeString,
 					Optional: true,
-					Default:  "Managed by Terraform",
+					Default:  "Managed by Pulumi",
 				},
 				"engine_patch_version": {
 					Type:     schema.TypeString,
diff --git a/internal/service/memorydb/parameter_group.go b/internal/service/memorydb/parameter_group.go
index 9976f6de38e..3a993a39483 100644
--- a/internal/service/memorydb/parameter_group.go
+++ b/internal/service/memorydb/parameter_group.go
@@ -51,7 +51,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/memorydb/subnet_group.go b/internal/service/memorydb/subnet_group.go
index 5bcc4b6868d..82246f48dcf 100644
--- a/internal/service/memorydb/subnet_group.go
+++ b/internal/service/memorydb/subnet_group.go
@@ -46,7 +46,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:          schema.TypeString,
diff --git a/internal/service/neptune/cluster_parameter_group.go b/internal/service/neptune/cluster_parameter_group.go
index 5d4d3b06a10..10c61ec5916 100644
--- a/internal/service/neptune/cluster_parameter_group.go
+++ b/internal/service/neptune/cluster_parameter_group.go
@@ -48,7 +48,7 @@ func resourceClusterParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/neptune/parameter_group.go b/internal/service/neptune/parameter_group.go
index 7fa8c36e688..da48a333793 100644
--- a/internal/service/neptune/parameter_group.go
+++ b/internal/service/neptune/parameter_group.go
@@ -54,7 +54,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/neptune/subnet_group.go b/internal/service/neptune/subnet_group.go
index d756af75db9..d448de51344 100644
--- a/internal/service/neptune/subnet_group.go
+++ b/internal/service/neptune/subnet_group.go
@@ -44,7 +44,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:          schema.TypeString,
diff --git a/internal/service/rds/cluster_parameter_group.go b/internal/service/rds/cluster_parameter_group.go
index 4c55067d087..6dcade3a407 100644
--- a/internal/service/rds/cluster_parameter_group.go
+++ b/internal/service/rds/cluster_parameter_group.go
@@ -50,7 +50,7 @@ func resourceClusterParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/rds/option_group.go b/internal/service/rds/option_group.go
index 821c3163eb9..ec05c242464 100644
--- a/internal/service/rds/option_group.go
+++ b/internal/service/rds/option_group.go
@@ -126,7 +126,7 @@ func resourceOptionGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrSkipDestroy: {
 				Type:     schema.TypeBool,
diff --git a/internal/service/rds/parameter_group.go b/internal/service/rds/parameter_group.go
index 100bd3bae4a..d9d02aea423 100644
--- a/internal/service/rds/parameter_group.go
+++ b/internal/service/rds/parameter_group.go
@@ -53,7 +53,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/rds/subnet_group.go b/internal/service/rds/subnet_group.go
index 6d6585687fb..1a3ce3438c1 100644
--- a/internal/service/rds/subnet_group.go
+++ b/internal/service/rds/subnet_group.go
@@ -47,7 +47,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:          schema.TypeString,
diff --git a/internal/service/redshift/parameter_group.go b/internal/service/redshift/parameter_group.go
index b712d3ec4b4..8c9f92ef8bf 100644
--- a/internal/service/redshift/parameter_group.go
+++ b/internal/service/redshift/parameter_group.go
@@ -50,7 +50,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/redshift/subnet_group.go b/internal/service/redshift/subnet_group.go
index 7117c8e0612..0fa362cf95a 100644
--- a/internal/service/redshift/subnet_group.go
+++ b/internal/service/redshift/subnet_group.go
@@ -45,7 +45,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:     schema.TypeString,
diff --git a/internal/service/route53/record.go b/internal/service/route53/record.go
index 10b95ec0c7f..9dba7752731 100644
--- a/internal/service/route53/record.go
+++ b/internal/service/route53/record.go
@@ -360,7 +360,7 @@ func resourceRecordCreate(ctx context.Context, d *schema.ResourceData, meta any)
 					ResourceRecordSet: expandResourceRecordSet(d, aws.ToString(zoneRecord.HostedZone.Name)),
 				},
 			},
-			Comment: aws.String("Managed by Terraform"),
+			Comment: aws.String("Managed by Pulumi"),
 		},
 		HostedZoneId: aws.String(cleanZoneID(aws.ToString(zoneRecord.HostedZone.Id))),
 	}
@@ -662,7 +662,7 @@ func resourceRecordUpdate(ctx context.Context, d *schema.ResourceData, meta any)
 					ResourceRecordSet: expandResourceRecordSet(d, aws.ToString(zoneRecord.HostedZone.Name)),
 				},
 			},
-			Comment: aws.String("Managed by Terraform"),
+			Comment: aws.String("Managed by Pulumi"),
 		},
 		HostedZoneId: aws.String(cleanZoneID(aws.ToString(zoneRecord.HostedZone.Id))),
 	}
@@ -720,7 +720,7 @@ func resourceRecordDelete(ctx context.Context, d *schema.ResourceData, meta any)
 					ResourceRecordSet: rec,
 				},
 			},
-			Comment: aws.String("Deleted by Terraform"),
+			Comment: aws.String("Deleted by Pulumi"),
 		},
 		HostedZoneId: aws.String(zoneID),
 	}
diff --git a/internal/service/route53/zone.go b/internal/service/route53/zone.go
index a8126d1ffa3..8ac90091bd4 100644
--- a/internal/service/route53/zone.go
+++ b/internal/service/route53/zone.go
@@ -51,7 +51,7 @@ func resourceZone() *schema.Resource {
 			names.AttrComment: {
 				Type:         schema.TypeString,
 				Optional:     true,
-				Default:      "Managed by Terraform",
+				Default:      "Managed by Pulumi",
 				ValidateFunc: validation.StringLenBetween(0, 256),
 			},
 			"delegation_set_id": {
diff --git a/internal/service/route53/zone_association.go b/internal/service/route53/zone_association.go
index e08f54104a8..7d31bed1e25 100644
--- a/internal/service/route53/zone_association.go
+++ b/internal/service/route53/zone_association.go
@@ -80,7 +80,7 @@ func resourceZoneAssociationCreate(ctx context.Context, d *schema.ResourceData,
 	zoneID := d.Get("zone_id").(string)
 	id := zoneAssociationCreateResourceID(zoneID, vpcID, vpcRegion)
 	input := &route53.AssociateVPCWithHostedZoneInput{
-		Comment:      aws.String("Managed by Terraform"),
+		Comment:      aws.String("Managed by Pulumi"),
 		HostedZoneId: aws.String(zoneID),
 		VPC: &awstypes.VPC{
 			VPCId:     aws.String(vpcID),
@@ -164,7 +164,7 @@ func resourceZoneAssociationDelete(ctx context.Context, d *schema.ResourceData,
 
 	log.Printf("[INFO] Deleting Route53 Zone Association: %s", d.Id())
 	output, err := conn.DisassociateVPCFromHostedZone(ctx, &route53.DisassociateVPCFromHostedZoneInput{
-		Comment:      aws.String("Managed by Terraform"),
+		Comment:      aws.String("Managed by Pulumi"),
 		HostedZoneId: aws.String(zoneID),
 		VPC: &awstypes.VPC{
 			VPCId:     aws.String(vpcID),
