From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: corymhall <43035978+corymhall@users.noreply.github.com>
Date: Wed, 29 May 2024 09:05:36 -0400
Subject: [PATCH] Change default descriptions to "Managed by Pulumi"


diff --git a/internal/service/apigateway/api_key.go b/internal/service/apigateway/api_key.go
index 0536b41089c..2ccd2c9beee 100644
--- a/internal/service/apigateway/api_key.go
+++ b/internal/service/apigateway/api_key.go
@@ -55,7 +55,7 @@ func resourceAPIKey() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrEnabled: {
 				Type:     schema.TypeBool,
diff --git a/internal/service/appsync/api_key.go b/internal/service/appsync/api_key.go
index 891c70f547c..11b9c2c6c46 100644
--- a/internal/service/appsync/api_key.go
+++ b/internal/service/appsync/api_key.go
@@ -52,7 +52,7 @@ func resourceAPIKey() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			"expires": {
 				Type:     schema.TypeString,
diff --git a/internal/service/cloudfront/origin_access_control.go b/internal/service/cloudfront/origin_access_control.go
index 4b9a6cd2aa6..affd26557ec 100644
--- a/internal/service/cloudfront/origin_access_control.go
+++ b/internal/service/cloudfront/origin_access_control.go
@@ -43,7 +43,7 @@ func resourceOriginAccessControl() *schema.Resource {
 			names.AttrDescription: {
 				Type:         schema.TypeString,
 				Optional:     true,
-				Default:      "Managed by Terraform",
+				Default:      "Managed by Pulumi",
 				ValidateFunc: validation.StringLenBetween(0, 256),
 			},
 			"etag": {
diff --git a/internal/service/docdb/cluster_parameter_group.go b/internal/service/docdb/cluster_parameter_group.go
index f60f08f9038..e1fec20a91d 100644
--- a/internal/service/docdb/cluster_parameter_group.go
+++ b/internal/service/docdb/cluster_parameter_group.go
@@ -50,7 +50,7 @@ func resourceClusterParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/docdb/subnet_group.go b/internal/service/docdb/subnet_group.go
index 8c8a796f6cb..8e72689323e 100644
--- a/internal/service/docdb/subnet_group.go
+++ b/internal/service/docdb/subnet_group.go
@@ -46,7 +46,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:          schema.TypeString,
diff --git a/internal/service/ec2/vpc_security_group.go b/internal/service/ec2/vpc_security_group.go
index 49a40b1577d..64d71c6008c 100644
--- a/internal/service/ec2/vpc_security_group.go
+++ b/internal/service/ec2/vpc_security_group.go
@@ -68,7 +68,7 @@ func resourceSecurityGroup() *schema.Resource {
 				Type:         schema.TypeString,
 				Optional:     true,
 				ForceNew:     true,
-				Default:      "Managed by Terraform",
+				Default:      "Managed by Pulumi",
 				ValidateFunc: validation.StringLenBetween(0, 255),
 			},
 			"egress":  securityGroupRuleSetNestedBlock,
diff --git a/internal/service/elasticache/parameter_group.go b/internal/service/elasticache/parameter_group.go
index cb33f88b44f..c260892165e 100644
--- a/internal/service/elasticache/parameter_group.go
+++ b/internal/service/elasticache/parameter_group.go
@@ -49,7 +49,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/elasticache/subnet_group.go b/internal/service/elasticache/subnet_group.go
index aa3dbe43b91..3d02a07f700 100644
--- a/internal/service/elasticache/subnet_group.go
+++ b/internal/service/elasticache/subnet_group.go
@@ -46,7 +46,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:     schema.TypeString,
diff --git a/internal/service/mediapackage/channel.go b/internal/service/mediapackage/channel.go
index 800c16f716b..e0ca1041fcc 100644
--- a/internal/service/mediapackage/channel.go
+++ b/internal/service/mediapackage/channel.go
@@ -50,7 +50,7 @@ func ResourceChannel() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			"hls_ingest": {
 				Type:     schema.TypeList,
diff --git a/internal/service/memorydb/cluster.go b/internal/service/memorydb/cluster.go
index eff7e54708c..b62ecc828ec 100644
--- a/internal/service/memorydb/cluster.go
+++ b/internal/service/memorydb/cluster.go
@@ -78,7 +78,7 @@ func resourceCluster() *schema.Resource {
 				names.AttrDescription: {
 					Type:     schema.TypeString,
 					Optional: true,
-					Default:  "Managed by Terraform",
+					Default:  "Managed by Pulumi",
 				},
 				"engine_patch_version": {
 					Type:     schema.TypeString,
diff --git a/internal/service/memorydb/parameter_group.go b/internal/service/memorydb/parameter_group.go
index ab552f51091..9261af83fd5 100644
--- a/internal/service/memorydb/parameter_group.go
+++ b/internal/service/memorydb/parameter_group.go
@@ -52,7 +52,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/memorydb/subnet_group.go b/internal/service/memorydb/subnet_group.go
index 5f9a456efa4..5a944346862 100644
--- a/internal/service/memorydb/subnet_group.go
+++ b/internal/service/memorydb/subnet_group.go
@@ -47,7 +47,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:          schema.TypeString,
diff --git a/internal/service/neptune/cluster_parameter_group.go b/internal/service/neptune/cluster_parameter_group.go
index 8f79776804b..b7cfba4803d 100644
--- a/internal/service/neptune/cluster_parameter_group.go
+++ b/internal/service/neptune/cluster_parameter_group.go
@@ -49,7 +49,7 @@ func resourceClusterParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/neptune/parameter_group.go b/internal/service/neptune/parameter_group.go
index 6eebfc93b15..7a9629497df 100644
--- a/internal/service/neptune/parameter_group.go
+++ b/internal/service/neptune/parameter_group.go
@@ -55,7 +55,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/neptune/subnet_group.go b/internal/service/neptune/subnet_group.go
index 7f633e06ac7..72d5cfd7621 100644
--- a/internal/service/neptune/subnet_group.go
+++ b/internal/service/neptune/subnet_group.go
@@ -45,7 +45,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:          schema.TypeString,
diff --git a/internal/service/rds/cluster_parameter_group.go b/internal/service/rds/cluster_parameter_group.go
index 4f38539432e..f2bab56a175 100644
--- a/internal/service/rds/cluster_parameter_group.go
+++ b/internal/service/rds/cluster_parameter_group.go
@@ -51,7 +51,7 @@ func resourceClusterParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/rds/option_group.go b/internal/service/rds/option_group.go
index 949d9038cd3..169590a155e 100644
--- a/internal/service/rds/option_group.go
+++ b/internal/service/rds/option_group.go
@@ -127,7 +127,7 @@ func resourceOptionGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrSkipDestroy: {
 				Type:     schema.TypeBool,
diff --git a/internal/service/rds/parameter_group.go b/internal/service/rds/parameter_group.go
index 3cf74395386..76545098f6d 100644
--- a/internal/service/rds/parameter_group.go
+++ b/internal/service/rds/parameter_group.go
@@ -54,7 +54,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/rds/subnet_group.go b/internal/service/rds/subnet_group.go
index 311a034b594..c35c3f60598 100644
--- a/internal/service/rds/subnet_group.go
+++ b/internal/service/rds/subnet_group.go
@@ -48,7 +48,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:          schema.TypeString,
diff --git a/internal/service/redshift/parameter_group.go b/internal/service/redshift/parameter_group.go
index a72f197bfd8..915faab4044 100644
--- a/internal/service/redshift/parameter_group.go
+++ b/internal/service/redshift/parameter_group.go
@@ -52,7 +52,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/redshift/subnet_group.go b/internal/service/redshift/subnet_group.go
index 9b7ff7826f2..f4284b7f312 100644
--- a/internal/service/redshift/subnet_group.go
+++ b/internal/service/redshift/subnet_group.go
@@ -46,7 +46,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:     schema.TypeString,
diff --git a/internal/service/route53/record.go b/internal/service/route53/record.go
index d3ad6cc9e66..46263569f4b 100644
--- a/internal/service/route53/record.go
+++ b/internal/service/route53/record.go
@@ -361,7 +361,7 @@ func resourceRecordCreate(ctx context.Context, d *schema.ResourceData, meta any)
 					ResourceRecordSet: expandResourceRecordSet(d, aws.ToString(zoneRecord.HostedZone.Name)),
 				},
 			},
-			Comment: aws.String("Managed by Terraform"),
+			Comment: aws.String("Managed by Pulumi"),
 		},
 		HostedZoneId: aws.String(cleanZoneID(aws.ToString(zoneRecord.HostedZone.Id))),
 	}
@@ -663,7 +663,7 @@ func resourceRecordUpdate(ctx context.Context, d *schema.ResourceData, meta any)
 					ResourceRecordSet: expandResourceRecordSet(d, aws.ToString(zoneRecord.HostedZone.Name)),
 				},
 			},
-			Comment: aws.String("Managed by Terraform"),
+			Comment: aws.String("Managed by Pulumi"),
 		},
 		HostedZoneId: aws.String(cleanZoneID(aws.ToString(zoneRecord.HostedZone.Id))),
 	}
@@ -721,7 +721,7 @@ func resourceRecordDelete(ctx context.Context, d *schema.ResourceData, meta any)
 					ResourceRecordSet: rec,
 				},
 			},
-			Comment: aws.String("Deleted by Terraform"),
+			Comment: aws.String("Deleted by Pulumi"),
 		},
 		HostedZoneId: aws.String(zoneID),
 	}
diff --git a/internal/service/route53/zone.go b/internal/service/route53/zone.go
index 87d979696b2..b3f7e0ead1f 100644
--- a/internal/service/route53/zone.go
+++ b/internal/service/route53/zone.go
@@ -52,7 +52,7 @@ func resourceZone() *schema.Resource {
 			names.AttrComment: {
 				Type:         schema.TypeString,
 				Optional:     true,
-				Default:      "Managed by Terraform",
+				Default:      "Managed by Pulumi",
 				ValidateFunc: validation.StringLenBetween(0, 256),
 			},
 			"delegation_set_id": {
diff --git a/internal/service/route53/zone_association.go b/internal/service/route53/zone_association.go
index 0a955a6968a..35644a36200 100644
--- a/internal/service/route53/zone_association.go
+++ b/internal/service/route53/zone_association.go
@@ -81,7 +81,7 @@ func resourceZoneAssociationCreate(ctx context.Context, d *schema.ResourceData,
 	zoneID := d.Get("zone_id").(string)
 	id := zoneAssociationCreateResourceID(zoneID, vpcID, vpcRegion)
 	input := &route53.AssociateVPCWithHostedZoneInput{
-		Comment:      aws.String("Managed by Terraform"),
+		Comment:      aws.String("Managed by Pulumi"),
 		HostedZoneId: aws.String(zoneID),
 		VPC: &awstypes.VPC{
 			VPCId:     aws.String(vpcID),
@@ -165,7 +165,7 @@ func resourceZoneAssociationDelete(ctx context.Context, d *schema.ResourceData,
 
 	log.Printf("[INFO] Deleting Route53 Zone Association: %s", d.Id())
 	output, err := conn.DisassociateVPCFromHostedZone(ctx, &route53.DisassociateVPCFromHostedZoneInput{
-		Comment:      aws.String("Managed by Terraform"),
+		Comment:      aws.String("Managed by Pulumi"),
 		HostedZoneId: aws.String(zoneID),
 		VPC: &awstypes.VPC{
 			VPCId:     aws.String(vpcID),
