From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Fri, 25 Oct 2024 17:14:36 -0400
Subject: [PATCH] Patch ACM retry to not retry after LimitExceededException
 (#3290)


diff --git a/internal/service/acm/service_package_extra.go b/internal/service/acm/service_package_extra.go
new file mode 100644
index 00000000000..b432bed386e
--- /dev/null
+++ b/internal/service/acm/service_package_extra.go
@@ -0,0 +1,21 @@
+package acm
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws"
+	retry_sdkv2 "github.com/aws/aws-sdk-go-v2/aws/retry"
+	"github.com/aws/aws-sdk-go-v2/service/acm"
+	tfawserr_sdkv2 "github.com/hashicorp/aws-sdk-go-base/v2/tfawserr"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func (p *servicePackage) pulumiCustomizeRetries(cfg aws.Config) func(*acm.Options) {
+	return func(o *acm.Options) {
+		o.Retryer = conns.AddIsErrorRetryables(cfg.Retryer().(aws.RetryerV2), retry_sdkv2.IsErrorRetryableFunc(func(err error) aws.Ternary {
+			if tfawserr_sdkv2.ErrMessageContains(err, "LimitExceededException", "the maximum number of") &&
+				tfawserr_sdkv2.ErrMessageContains(err, "LimitExceededException", "certificates in the last year") {
+				return aws.FalseTernary
+			}
+			return aws.UnknownTernary // Delegate to configured Retryer.
+		}))
+	}
+}
diff --git a/internal/service/acm/service_package_gen.go b/internal/service/acm/service_package_gen.go
index b037e6f9000..565745b8c95 100644
--- a/internal/service/acm/service_package_gen.go
+++ b/internal/service/acm/service_package_gen.go
@@ -93,6 +93,7 @@ func (p *servicePackage) NewClient(ctx context.Context, config map[string]any) (
 			}
 		},
 		withExtraOptions(ctx, p, config),
+		p.pulumiCustomizeRetries(cfg),
 	}
 
 	return acm.NewFromConfig(cfg, optFns...), nil
