From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: corymhall <43035978+corymhall@users.noreply.github.com>
Date: Thu, 11 Apr 2024 14:42:01 -0400
Subject: [PATCH] non-idempotent sns topic creation


diff --git a/internal/service/sns/topic.go b/internal/service/sns/topic.go
index 7f39305a82..1b97fccfe4 100644
--- a/internal/service/sns/topic.go
+++ b/internal/service/sns/topic.go
@@ -10,6 +10,7 @@ import (
 	"log"
 	"regexp"
 	"strconv"
+	"sync"
 	"time"
 
 	"github.com/YakDriver/regexache"
@@ -263,6 +264,12 @@ func resourceTopic() *schema.Resource {
 	}
 }
 
+func constructTopicArn(client *sns.Client, account, region, partition, snsTopicName string) string {
+	return fmt.Sprintf("arn:%s:sns:%s:%s:%s", partition, region, account, snsTopicName)
+}
+
+var snsGlobalMutex sync.Map
+
 func resourceTopicCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	var diags diag.Diagnostics
 	conn := meta.(*conns.AWSClient).SNSClient(ctx)
@@ -291,6 +298,28 @@ func resourceTopicCreate(ctx context.Context, d *schema.ResourceData, meta any)
 		delete(attributes, topicAttributeNameFIFOThroughputScope)
 	}
 
+	// create a lock based on the topic ARN. We really want to make sure
+	// that we prevent a race condition where two resources are created with
+	// the same name.
+	awsClient := meta.(*conns.AWSClient)
+	topicArn := constructTopicArn(conn, awsClient.AccountID(ctx), awsClient.Region(ctx), awsClient.Partition(ctx), name)
+	localMutex := &sync.Mutex{}
+	if val, ok := snsGlobalMutex.LoadOrStore(topicArn, localMutex); ok {
+		localMutex = val.(*sync.Mutex)
+	}
+	localMutex.Lock()
+	defer localMutex.Unlock()
+
+	// Look up if the topic already exists
+	_, err = findTopicAttributesWithValidAWSPrincipalsByARN(ctx, conn, topicArn)
+
+	if err == nil {
+		return diag.Errorf("SNS Topic (%s) already exists", name)
+	}
+	if !tfresource.NotFound(err) {
+		return diag.FromErr(err)
+	}
+
 	output, err := conn.CreateTopic(ctx, input)
 
 	// Some partitions (e.g. ISO) may not support tag-on-create.
