From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: corymhall <43035978+corymhall@users.noreply.github.com>
Date: Wed, 29 May 2024 09:05:36 -0400
Subject: [PATCH] Change default descriptions to "Managed by Pulumi"


diff --git a/internal/service/apigateway/api_key.go b/internal/service/apigateway/api_key.go
index 6351d6ff42..6c968b3007 100644
--- a/internal/service/apigateway/api_key.go
+++ b/internal/service/apigateway/api_key.go
@@ -55,7 +55,7 @@ func resourceAPIKey() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrEnabled: {
 				Type:     schema.TypeBool,
diff --git a/internal/service/appsync/api_key.go b/internal/service/appsync/api_key.go
index fbdca7da91..9cdc01f8ec 100644
--- a/internal/service/appsync/api_key.go
+++ b/internal/service/appsync/api_key.go
@@ -49,7 +49,7 @@ func resourceAPIKey() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			"expires": {
 				Type:     schema.TypeString,
diff --git a/internal/service/cloudfront/origin_access_control.go b/internal/service/cloudfront/origin_access_control.go
index 837e79c057..8fc8280900 100644
--- a/internal/service/cloudfront/origin_access_control.go
+++ b/internal/service/cloudfront/origin_access_control.go
@@ -38,7 +38,7 @@ func resourceOriginAccessControl() *schema.Resource {
 			names.AttrDescription: {
 				Type:         schema.TypeString,
 				Optional:     true,
-				Default:      "Managed by Terraform",
+				Default:      "Managed by Pulumi",
 				ValidateFunc: validation.StringLenBetween(0, 256),
 			},
 			"etag": {
diff --git a/internal/service/docdb/cluster_parameter_group.go b/internal/service/docdb/cluster_parameter_group.go
index 8a99e5b047..361c39f731 100644
--- a/internal/service/docdb/cluster_parameter_group.go
+++ b/internal/service/docdb/cluster_parameter_group.go
@@ -50,7 +50,7 @@ func resourceClusterParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/docdb/subnet_group.go b/internal/service/docdb/subnet_group.go
index 5466936547..526c40e604 100644
--- a/internal/service/docdb/subnet_group.go
+++ b/internal/service/docdb/subnet_group.go
@@ -46,7 +46,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:          schema.TypeString,
diff --git a/internal/service/ec2/vpc_security_group.go b/internal/service/ec2/vpc_security_group.go
index 82b8dae6e7..a89db0b86d 100644
--- a/internal/service/ec2/vpc_security_group.go
+++ b/internal/service/ec2/vpc_security_group.go
@@ -69,7 +69,7 @@ func resourceSecurityGroup() *schema.Resource {
 				Type:         schema.TypeString,
 				Optional:     true,
 				ForceNew:     true,
-				Default:      "Managed by Terraform",
+				Default:      "Managed by Pulumi",
 				ValidateFunc: validation.StringLenBetween(0, 255),
 			},
 			"egress":  securityGroupRuleSetNestedBlock,
diff --git a/internal/service/elasticache/parameter_group.go b/internal/service/elasticache/parameter_group.go
index 5a265a26d8..ecc1f2d6ac 100644
--- a/internal/service/elasticache/parameter_group.go
+++ b/internal/service/elasticache/parameter_group.go
@@ -49,7 +49,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/elasticache/subnet_group.go b/internal/service/elasticache/subnet_group.go
index 89920a47e3..71f4a900d7 100644
--- a/internal/service/elasticache/subnet_group.go
+++ b/internal/service/elasticache/subnet_group.go
@@ -48,7 +48,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:     schema.TypeString,
diff --git a/internal/service/mediapackage/channel.go b/internal/service/mediapackage/channel.go
index 961d1f0438..710fd0b998 100644
--- a/internal/service/mediapackage/channel.go
+++ b/internal/service/mediapackage/channel.go
@@ -51,7 +51,7 @@ func ResourceChannel() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			"hls_ingest": {
 				Type:     schema.TypeList,
diff --git a/internal/service/memorydb/cluster.go b/internal/service/memorydb/cluster.go
index f3a35c84e5..1e3027048a 100644
--- a/internal/service/memorydb/cluster.go
+++ b/internal/service/memorydb/cluster.go
@@ -79,7 +79,7 @@ func resourceCluster() *schema.Resource {
 				names.AttrDescription: {
 					Type:     schema.TypeString,
 					Optional: true,
-					Default:  "Managed by Terraform",
+					Default:  "Managed by Pulumi",
 				},
 				"engine_patch_version": {
 					Type:     schema.TypeString,
diff --git a/internal/service/memorydb/parameter_group.go b/internal/service/memorydb/parameter_group.go
index 625357ac1b..c10f1daa43 100644
--- a/internal/service/memorydb/parameter_group.go
+++ b/internal/service/memorydb/parameter_group.go
@@ -54,7 +54,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/memorydb/subnet_group.go b/internal/service/memorydb/subnet_group.go
index b01015569f..1e3d84acfa 100644
--- a/internal/service/memorydb/subnet_group.go
+++ b/internal/service/memorydb/subnet_group.go
@@ -49,7 +49,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:          schema.TypeString,
diff --git a/internal/service/neptune/cluster_parameter_group.go b/internal/service/neptune/cluster_parameter_group.go
index b31cd660ef..8ec5d31fae 100644
--- a/internal/service/neptune/cluster_parameter_group.go
+++ b/internal/service/neptune/cluster_parameter_group.go
@@ -49,7 +49,7 @@ func resourceClusterParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/neptune/parameter_group.go b/internal/service/neptune/parameter_group.go
index df81d7c11b..518b7c2f11 100644
--- a/internal/service/neptune/parameter_group.go
+++ b/internal/service/neptune/parameter_group.go
@@ -55,7 +55,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/neptune/subnet_group.go b/internal/service/neptune/subnet_group.go
index 1d97895eea..45e74ae9ce 100644
--- a/internal/service/neptune/subnet_group.go
+++ b/internal/service/neptune/subnet_group.go
@@ -45,7 +45,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:          schema.TypeString,
diff --git a/internal/service/rds/cluster_parameter_group.go b/internal/service/rds/cluster_parameter_group.go
index 28a4a0c6ae..26112e8db5 100644
--- a/internal/service/rds/cluster_parameter_group.go
+++ b/internal/service/rds/cluster_parameter_group.go
@@ -51,7 +51,7 @@ func resourceClusterParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/rds/option_group.go b/internal/service/rds/option_group.go
index d19358fa76..b13859ab8e 100644
--- a/internal/service/rds/option_group.go
+++ b/internal/service/rds/option_group.go
@@ -127,7 +127,7 @@ func resourceOptionGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrSkipDestroy: {
 				Type:     schema.TypeBool,
diff --git a/internal/service/rds/parameter_group.go b/internal/service/rds/parameter_group.go
index 0e4ca3f870..c1e331f54f 100644
--- a/internal/service/rds/parameter_group.go
+++ b/internal/service/rds/parameter_group.go
@@ -53,7 +53,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/rds/subnet_group.go b/internal/service/rds/subnet_group.go
index 646add5ad8..2731f4e425 100644
--- a/internal/service/rds/subnet_group.go
+++ b/internal/service/rds/subnet_group.go
@@ -48,7 +48,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:          schema.TypeString,
diff --git a/internal/service/redshift/parameter_group.go b/internal/service/redshift/parameter_group.go
index 0552e455d6..d413e1baf8 100644
--- a/internal/service/redshift/parameter_group.go
+++ b/internal/service/redshift/parameter_group.go
@@ -51,7 +51,7 @@ func resourceParameterGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrFamily: {
 				Type:     schema.TypeString,
diff --git a/internal/service/redshift/subnet_group.go b/internal/service/redshift/subnet_group.go
index d95746d343..47390e9d5b 100644
--- a/internal/service/redshift/subnet_group.go
+++ b/internal/service/redshift/subnet_group.go
@@ -46,7 +46,7 @@ func resourceSubnetGroup() *schema.Resource {
 			names.AttrDescription: {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "Managed by Terraform",
+				Default:  "Managed by Pulumi",
 			},
 			names.AttrName: {
 				Type:     schema.TypeString,
diff --git a/internal/service/route53/record.go b/internal/service/route53/record.go
index c3170d4965..f13dada174 100644
--- a/internal/service/route53/record.go
+++ b/internal/service/route53/record.go
@@ -360,7 +360,7 @@ func resourceRecordCreate(ctx context.Context, d *schema.ResourceData, meta inte
 					ResourceRecordSet: expandResourceRecordSet(d, aws.ToString(zoneRecord.HostedZone.Name)),
 				},
 			},
-			Comment: aws.String("Managed by Terraform"),
+			Comment: aws.String("Managed by Pulumi"),
 		},
 		HostedZoneId: aws.String(cleanZoneID(aws.ToString(zoneRecord.HostedZone.Id))),
 	}
@@ -670,7 +670,7 @@ func resourceRecordUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 					ResourceRecordSet: expandResourceRecordSet(d, aws.ToString(zoneRecord.HostedZone.Name)),
 				},
 			},
-			Comment: aws.String("Managed by Terraform"),
+			Comment: aws.String("Managed by Pulumi"),
 		},
 		HostedZoneId: aws.String(cleanZoneID(aws.ToString(zoneRecord.HostedZone.Id))),
 	}
@@ -737,7 +737,7 @@ func resourceRecordDelete(ctx context.Context, d *schema.ResourceData, meta inte
 					ResourceRecordSet: rec,
 				},
 			},
-			Comment: aws.String("Deleted by Terraform"),
+			Comment: aws.String("Deleted by Pulumi"),
 		},
 		HostedZoneId: aws.String(zoneID),
 	}
diff --git a/internal/service/route53/zone.go b/internal/service/route53/zone.go
index 86b843b763..21ed69c593 100644
--- a/internal/service/route53/zone.go
+++ b/internal/service/route53/zone.go
@@ -52,7 +52,7 @@ func resourceZone() *schema.Resource {
 			names.AttrComment: {
 				Type:         schema.TypeString,
 				Optional:     true,
-				Default:      "Managed by Terraform",
+				Default:      "Managed by Pulumi",
 				ValidateFunc: validation.StringLenBetween(0, 256),
 			},
 			"delegation_set_id": {
diff --git a/internal/service/route53/zone_association.go b/internal/service/route53/zone_association.go
index 6bf4c1ef12..19c08a7250 100644
--- a/internal/service/route53/zone_association.go
+++ b/internal/service/route53/zone_association.go
@@ -74,7 +74,7 @@ func resourceZoneAssociationCreate(ctx context.Context, d *schema.ResourceData,
 	zoneID := d.Get("zone_id").(string)
 	id := zoneAssociationCreateResourceID(zoneID, vpcID, vpcRegion)
 	input := &route53.AssociateVPCWithHostedZoneInput{
-		Comment:      aws.String("Managed by Terraform"),
+		Comment:      aws.String("Managed by Pulumi"),
 		HostedZoneId: aws.String(zoneID),
 		VPC: &awstypes.VPC{
 			VPCId:     aws.String(vpcID),
@@ -158,7 +158,7 @@ func resourceZoneAssociationDelete(ctx context.Context, d *schema.ResourceData,
 
 	log.Printf("[INFO] Deleting Route53 Zone Association: %s", d.Id())
 	output, err := conn.DisassociateVPCFromHostedZone(ctx, &route53.DisassociateVPCFromHostedZoneInput{
-		Comment:      aws.String("Managed by Terraform"),
+		Comment:      aws.String("Managed by Pulumi"),
 		HostedZoneId: aws.String(zoneID),
 		VPC: &awstypes.VPC{
 			VPCId:     aws.String(vpcID),
