From 92be6c8ed60b315474cdac19726012a38b5ddd0a Mon Sep 17 00:00:00 2001
From: Daniel Bradley <daniel@pulumi.com>
Date: Thu, 9 Mar 2023 14:50:36 +0000
Subject: [PATCH 20/30] Fixup eks formatting

---
 internal/service/eks/wait.go | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/internal/service/eks/wait.go b/internal/service/eks/wait.go
index 6568e06549..fb1ec98588 100644
--- a/internal/service/eks/wait.go
+++ b/internal/service/eks/wait.go
@@ -14,9 +14,9 @@ import (
 )
 
 const (
-	addonCreatedTimeout = 20 * time.Minute
-	addonUpdatedTimeout = 20 * time.Minute
-	addonDeletedTimeout = 40 * time.Minute
+	addonCreatedTimeout       = 20 * time.Minute
+	addonUpdatedTimeout       = 20 * time.Minute
+	addonDeletedTimeout       = 40 * time.Minute
 	clusterDeleteRetryTimeout = 60 * time.Minute
 )
 
-- 
2.40.0

