From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Guinevere Saenger <guinevere@pulumi.com>
Date: Wed, 6 Sep 2023 10:43:30 -0700
Subject: [PATCH] move shim logic to upstream as a patch


diff --git a/shim/shim.go b/shim/shim.go
new file mode 100644
index 00000000000..2af7c069252
--- /dev/null
+++ b/shim/shim.go
@@ -0,0 +1,36 @@
+package shim
+
+import (
+	"context"
+
+	pfprovider "github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/provider"
+	"github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider"
+	"github.com/hashicorp/terraform-provider-aws/internal/tags"
+)
+
+type UpstreamProvider struct {
+	SDKV2Provider           *schema.Provider
+	PluginFrameworkProvider pfprovider.Provider
+}
+
+func NewUpstreamProvider(ctx context.Context) (UpstreamProvider, error) {
+	primary, err := provider.New(ctx)
+	if err != nil {
+		return UpstreamProvider{}, err
+	}
+	pf := fwprovider.New(primary)
+	return UpstreamProvider{
+		SDKV2Provider:           primary,
+		PluginFrameworkProvider: pf,
+	}, nil
+}
+
+type TagConfig = tags.DefaultConfig
+
+type TagIgnoreConfig = tags.IgnoreConfig
+
+func NewTagConfig(ctx context.Context, i interface{}) TagConfig {
+	return TagConfig{Tags: tags.New(ctx, i)}
+}
