From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: corymhall <43035978+corymhall@users.noreply.github.com>
Date: Tue, 9 Apr 2024 15:24:26 -0400
Subject: [PATCH] Restore S3ConnURICleaningDisabled


diff --git a/internal/conns/awsclient.go b/internal/conns/awsclient.go
index 8f0be787649..627fbb1da16 100644
--- a/internal/conns/awsclient.go
+++ b/internal/conns/awsclient.go
@@ -208,7 +208,7 @@ func (c *AWSClient) DefaultKMSKeyPolicy(ctx context.Context) string {
 			"Resource": "*"
 		}
 	]
-}	
+}
 `, c.Partition(ctx), c.AccountID)
 }
 
diff --git a/internal/conns/awsclient_extra.go b/internal/conns/awsclient_extra.go
new file mode 100644
index 00000000000..96c809fc475
--- /dev/null
+++ b/internal/conns/awsclient_extra.go
@@ -0,0 +1,22 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package conns
+
+import (
+	"context"
+
+	s3_sdkv1 "github.com/aws/aws-sdk-go/service/s3"
+)
+
+func (c *AWSClient) S3Conn(ctx context.Context) *s3_sdkv1.S3 {
+	return errs.Must(conn[*s3_sdkv1.S3](ctx, c, names.S3, make(map[string]any)))
+}
+
+func (client *AWSClient) S3ConnURICleaningDisabled(ctx context.Context) *s3_sdkv1.S3 {
+	config := client.S3Conn(ctx).Config
+	t := true
+	config.DisableRestProtocolURICleaning = &t
+
+	return s3_sdkv1.New(client.session.Copy(&config))
+}
