From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: corymhall <43035978+corymhall@users.noreply.github.com>
Date: Tue, 9 Apr 2024 15:24:26 -0400
Subject: [PATCH 26/52] Restore S3ConnURICleaningDisabled


diff --git a/internal/conns/awsclient.go b/internal/conns/awsclient.go
index bef13bbc88..73b2147f73 100644
--- a/internal/conns/awsclient.go
+++ b/internal/conns/awsclient.go
@@ -23,6 +23,7 @@ import (
 	kms_sdkv1 "github.com/aws/aws-sdk-go/service/kms"
 	opsworks_sdkv1 "github.com/aws/aws-sdk-go/service/opsworks"
 	rds_sdkv1 "github.com/aws/aws-sdk-go/service/rds"
+	s3_sdkv1 "github.com/aws/aws-sdk-go/service/s3"
 	baselogging "github.com/hashicorp/aws-sdk-go-base/v2/logging"
 	"github.com/hashicorp/terraform-provider-aws/internal/errs"
 	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
@@ -165,6 +166,13 @@ func (c *AWSClient) S3UsePathStyle(context.Context) bool {
 	return c.s3UsePathStyle
 }
 
+func (client *AWSClient) S3ConnURICleaningDisabled(ctx context.Context) *s3_sdkv1.S3 {
+	config := client.S3Conn(ctx).Config
+	config.DisableRestProtocolURICleaning = aws_sdkv2.Bool(true)
+
+	return s3_sdkv1.New(client.session.Copy(&config))
+}
+
 // SetHTTPClient sets the http.Client used for AWS API calls.
 // To have effect it must be called before the AWS SDK v1 Session is created.
 func (c *AWSClient) SetHTTPClient(_ context.Context, httpClient *http.Client) {
@@ -230,7 +238,7 @@ func (c *AWSClient) DefaultKMSKeyPolicy(context.Context) string {
 			"Resource": "*"
 		}
 	]
-}	
+}
 `, c.Partition, c.AccountID)
 }
 
