From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bryce Lampe <bryce@pulumi.com>
Date: Fri, 13 Dec 2024 13:44:02 -0800
Subject: [PATCH] Speed up providerlint by re-using build cache and ignoring
 tests


diff --git a/GNUmakefile b/GNUmakefile
index 927014b500..3992e42178 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -346,9 +346,10 @@ prereq-go: ## If $(GO_VER) is not installed, install it
 
 provider-lint: ## [CI] ProviderLint Checks / providerlint
 	@echo "make: ProviderLint Checks / providerlint..."
-	@cd .ci/providerlint && go install -buildvcs=false .
-	@providerlint \
+	@cd .ci/providerlint && go build -buildvcs=false .
+	@.ci/providerlint/providerlint \
 		-c 1 \
+		-test=false \
 		-AT001.ignored-filename-suffixes=_data_source_test.go \
 		-AWSAT006=false \
 		-AWSR002=false \
