From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Graham Davison <gdavison@hashicorp.com>
Date: Thu, 28 May 2026 11:58:05 -0700
Subject: [PATCH] Excludes `semgrep` directory from `gofmt` checks


diff --git a/.ci/scripts/gofmtcheck.sh b/.ci/scripts/gofmtcheck.sh
index 0f25b4d6ebc..e1fc921e627 100755
--- a/.ci/scripts/gofmtcheck.sh
+++ b/.ci/scripts/gofmtcheck.sh
@@ -4,7 +4,7 @@

 # Check gofmt
 echo "==> Checking that code complies with gofmt requirements..."
-gofmt_files=$(find . -name '*.go' | grep -v vendor | xargs gofmt -l -s)
+gofmt_files=$(find . -name '*.go' | grep -v vendor | grep -v semgrep | xargs gofmt -l -s)
 if [[ -n ${gofmt_files} ]]; then
     echo 'gofmt needs running on the following files:'
     echo "${gofmt_files}"
