From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Wed, 6 Dec 2023 23:41:21 -0500
Subject: [PATCH 31/52] Fix job queue sdkv2 migration


diff --git a/internal/service/batch/job_queue_schema.go b/internal/service/batch/job_queue_schema.go
index 0ded1f6052..90de7f202f 100644
--- a/internal/service/batch/job_queue_schema.go
+++ b/internal/service/batch/job_queue_schema.go
@@ -93,6 +93,7 @@ func upgradeJobQueueResourceStateV0toV1(ctx context.Context, req resource.Upgrad
 	ceo := fwtypes.NewListNestedObjectValueOfNull[computeEnvironmentOrder](ctx)
 
 	jobQueueDataV2 := resourceJobQueueData{
+		ARN:                     jobQueueDataV0.ARN,
 		ComputeEnvironments:     jobQueueDataV0.ComputeEnvironments,
 		ComputeEnvironmentOrder: ceo,
 		ID:                      jobQueueDataV0.ID,
