From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Thu, 7 Dec 2023 00:05:40 -0500
Subject: [PATCH] Run scripts/patch_computed_only.sh to patch
 eks/pod_identity_association and more


diff --git a/internal/service/docdbelastic/cluster.go b/internal/service/docdbelastic/cluster.go
index 2d0c9e9114..1834e8e258 100644
--- a/internal/service/docdbelastic/cluster.go
+++ b/internal/service/docdbelastic/cluster.go
@@ -143,7 +143,7 @@ func (r *resourceCluster) Schema(ctx context.Context, _ resource.SchemaRequest,
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			names.AttrVPCSecurityGroupIDs: schema.SetAttribute{
 				CustomType: fwtypes.SetOfStringType,
 				Optional:   true,
diff --git a/internal/service/eks/pod_identity_association.go b/internal/service/eks/pod_identity_association.go
index 58e5a9fa34..3c1c3ca6d0 100644
--- a/internal/service/eks/pod_identity_association.go
+++ b/internal/service/eks/pod_identity_association.go
@@ -101,7 +101,7 @@ func (r *podIdentityAssociationResource) Schema(ctx context.Context, req resourc
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/s3control/access_grant.go b/internal/service/s3control/access_grant.go
index 6ba18f627c..56ea3d4aef 100644
--- a/internal/service/s3control/access_grant.go
+++ b/internal/service/s3control/access_grant.go
@@ -98,7 +98,7 @@ func (r *accessGrantResource) Schema(ctx context.Context, request resource.Schem
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"access_grants_location_configuration": schema.ListNestedBlock{
diff --git a/internal/service/s3control/access_grants_instance.go b/internal/service/s3control/access_grants_instance.go
index 57bc880e57..70981a8c02 100644
--- a/internal/service/s3control/access_grants_instance.go
+++ b/internal/service/s3control/access_grants_instance.go
@@ -78,7 +78,7 @@ func (r *accessGrantsInstanceResource) Schema(ctx context.Context, request resou
 				Optional:   true,
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/s3control/access_grants_location.go b/internal/service/s3control/access_grants_location.go
index 3707248eef..41c28a8503 100644
--- a/internal/service/s3control/access_grants_location.go
+++ b/internal/service/s3control/access_grants_location.go
@@ -79,7 +79,7 @@ func (r *accessGrantsLocationResource) Schema(ctx context.Context, request resou
 			},
 			names.AttrID:      framework.IDAttribute(),
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
