From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Wed, 6 Dec 2023 23:41:21 -0500
Subject: [PATCH 32/45] Fix job queue sdkv2 migration


diff --git a/internal/service/batch/job_queue_schema.go b/internal/service/batch/job_queue_schema.go
index 45c6b8cadc..11cf093ece 100644
--- a/internal/service/batch/job_queue_schema.go
+++ b/internal/service/batch/job_queue_schema.go
@@ -92,6 +92,7 @@ func upgradeJobQueueResourceStateV0toV1(ctx context.Context, req resource.Upgrad
 	}
 
 	jobQueueDataV2 := resourceJobQueueData{
+		ARN:                 jobQueueDataV0.ARN,
 		ComputeEnvironments: jobQueueDataV0.ComputeEnvironments,
 		ID:                  jobQueueDataV0.ID,
 		Name:                jobQueueDataV0.Name,
