From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Thu, 7 Dec 2023 00:05:40 -0500
Subject: [PATCH] Run scripts/patch_computed_only.sh to patch
 eks/pod_identity_association and more


diff --git a/internal/service/docdbelastic/cluster.go b/internal/service/docdbelastic/cluster.go
index 31b2fbe18f..aa6820adf4 100644
--- a/internal/service/docdbelastic/cluster.go
+++ b/internal/service/docdbelastic/cluster.go
@@ -127,7 +127,7 @@ func (r *resourceCluster) Schema(ctx context.Context, _ resource.SchemaRequest,
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"vpc_security_group_ids": schema.SetAttribute{
 				ElementType: types.StringType,
 				Optional:    true,
diff --git a/internal/service/eks/pod_identity_association.go b/internal/service/eks/pod_identity_association.go
index e4f57d15e7..672d9018c5 100644
--- a/internal/service/eks/pod_identity_association.go
+++ b/internal/service/eks/pod_identity_association.go
@@ -105,7 +105,7 @@ func (r *podIdentityAssociationResource) Schema(ctx context.Context, req resourc
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/s3control/access_grant.go b/internal/service/s3control/access_grant.go
index f2963c92a6..d93b539ff2 100644
--- a/internal/service/s3control/access_grant.go
+++ b/internal/service/s3control/access_grant.go
@@ -103,7 +103,7 @@ func (r *accessGrantResource) Schema(ctx context.Context, request resource.Schem
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"access_grants_location_configuration": schema.ListNestedBlock{
diff --git a/internal/service/s3control/access_grants_instance.go b/internal/service/s3control/access_grants_instance.go
index 8adcc8ce67..ceb5336ce9 100644
--- a/internal/service/s3control/access_grants_instance.go
+++ b/internal/service/s3control/access_grants_instance.go
@@ -82,7 +82,7 @@ func (r *accessGrantsInstanceResource) Schema(ctx context.Context, request resou
 				Optional:   true,
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/s3control/access_grants_location.go b/internal/service/s3control/access_grants_location.go
index f45d59078b..bead9b4bb9 100644
--- a/internal/service/s3control/access_grants_location.go
+++ b/internal/service/s3control/access_grants_location.go
@@ -84,7 +84,7 @@ func (r *accessGrantsLocationResource) Schema(ctx context.Context, request resou
 			},
 			names.AttrID:      framework.IDAttribute(),
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
