From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Thu, 7 Dec 2023 00:18:14 -0500
Subject: [PATCH] Fail fast when PF resources are dropped


diff --git a/internal/provider/fwprovider/provider.go b/internal/provider/fwprovider/provider.go
index a64bcd031d..0d730c8ea4 100644
--- a/internal/provider/fwprovider/provider.go
+++ b/internal/provider/fwprovider/provider.go
@@ -453,9 +453,8 @@ func (p *fwprovider) Resources(ctx context.Context) []func() resource.Resource {
 	}
 
 	if err := errors.Join(errs...); err != nil {
-		tflog.Warn(ctx, "registering resources", map[string]interface{}{
-			"error": err.Error(),
-		})
+		//lintignore:R009
+		panic(err)
 	}
 
 	return resources
