From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Wed, 7 Feb 2024 12:24:44 -0500
Subject: [PATCH] Fix tags_all Computed for PF resources


diff --git a/internal/service/amp/scraper.go b/internal/service/amp/scraper.go
index 1e7b516880..a0e509ab99 100644
--- a/internal/service/amp/scraper.go
+++ b/internal/service/amp/scraper.go
@@ -76,7 +76,7 @@ func (r *scraperResource) Schema(ctx context.Context, request resource.SchemaReq
 				Required: true,
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			names.AttrDestination: schema.ListNestedBlock{
diff --git a/internal/service/bedrock/custom_model.go b/internal/service/bedrock/custom_model.go
index 8ceb2c254c..c0dd56e718 100644
--- a/internal/service/bedrock/custom_model.go
+++ b/internal/service/bedrock/custom_model.go
@@ -135,7 +135,7 @@ func (r *customModelResource) Schema(ctx context.Context, request resource.Schem
 				},
 			},
 			names.AttrTags:       tftags.TagsAttribute(),
-			names.AttrTagsAll:    tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll:    tftags.TagsAttribute(),
 			"training_metrics":   framework.ResourceComputedListOfObjectsAttribute[trainingMetricsModel](ctx),
 			"validation_metrics": framework.ResourceComputedListOfObjectsAttribute[validatorMetricModel](ctx),
 		},
diff --git a/internal/service/codeguruprofiler/profiling_group.go b/internal/service/codeguruprofiler/profiling_group.go
index ba3d204d75..b16fc33f0b 100644
--- a/internal/service/codeguruprofiler/profiling_group.go
+++ b/internal/service/codeguruprofiler/profiling_group.go
@@ -69,7 +69,7 @@ func (r *resourceProfilingGroup) Schema(ctx context.Context, req resource.Schema
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"agent_orchestration_config": schema.ListNestedBlock{
diff --git a/internal/service/elasticache/serverless_cache.go b/internal/service/elasticache/serverless_cache.go
index cc5c312598..f453619c2c 100644
--- a/internal/service/elasticache/serverless_cache.go
+++ b/internal/service/elasticache/serverless_cache.go
@@ -181,7 +181,7 @@ func (r *serverlessCacheResource) Schema(ctx context.Context, request resource.S
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"user_group_id": schema.StringAttribute{
 				Optional: true,
 			},
diff --git a/internal/service/rekognition/collection.go b/internal/service/rekognition/collection.go
index 5561f7ecff..07a3c8d32d 100644
--- a/internal/service/rekognition/collection.go
+++ b/internal/service/rekognition/collection.go
@@ -73,7 +73,7 @@ func (r *resourceCollection) Schema(ctx context.Context, req resource.SchemaRequ
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 
diff --git a/internal/service/securityhub/automation_rule.go b/internal/service/securityhub/automation_rule.go
index 6360711508..75e0a82ccb 100644
--- a/internal/service/securityhub/automation_rule.go
+++ b/internal/service/securityhub/automation_rule.go
@@ -74,7 +74,7 @@ func (r *automationRuleResource) Schema(ctx context.Context, request resource.Sc
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			names.AttrActions: schema.SetNestedBlock{
diff --git a/internal/service/securitylake/data_lake.go b/internal/service/securitylake/data_lake.go
index c639d89b86..c4acedf082 100644
--- a/internal/service/securitylake/data_lake.go
+++ b/internal/service/securitylake/data_lake.go
@@ -71,7 +71,7 @@ func (r *dataLakeResource) Schema(ctx context.Context, request resource.SchemaRe
 			},
 			"s3_bucket_arn":   framework.ARNAttributeComputedOnly(),
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			names.AttrConfiguration: schema.ListNestedBlock{
diff --git a/internal/service/ssmcontacts/rotation.go b/internal/service/ssmcontacts/rotation.go
index 8c90e48854..ff5c702ea9 100644
--- a/internal/service/ssmcontacts/rotation.go
+++ b/internal/service/ssmcontacts/rotation.go
@@ -68,7 +68,7 @@ func (r *resourceRotation) Schema(ctx context.Context, request resource.SchemaRe
 				Optional:   true,
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"time_zone_id": schema.StringAttribute{
 				Required: true,
 			},
diff --git a/internal/service/ssoadmin/application.go b/internal/service/ssoadmin/application.go
index c339159fbe..797b807ed3 100644
--- a/internal/service/ssoadmin/application.go
+++ b/internal/service/ssoadmin/application.go
@@ -92,7 +92,7 @@ func (r *resourceApplication) Schema(ctx context.Context, req resource.SchemaReq
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"portal_options": schema.ListNestedBlock{
diff --git a/internal/service/ssoadmin/trusted_token_issuer.go b/internal/service/ssoadmin/trusted_token_issuer.go
index 28ce5529de..c462b619c6 100644
--- a/internal/service/ssoadmin/trusted_token_issuer.go
+++ b/internal/service/ssoadmin/trusted_token_issuer.go
@@ -82,7 +82,7 @@ func (r *resourceTrustedTokenIssuer) Schema(ctx context.Context, req resource.Sc
 			},
 
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"trusted_token_issuer_configuration": schema.ListNestedBlock{
