diff --git a/internal/service/sns/topic.go b/internal/service/sns/topic.go
index 98ebd87d5d..fa76769358 100644
--- a/internal/service/sns/topic.go
+++ b/internal/service/sns/topic.go
@@ -9,6 +9,7 @@ import (
 	"fmt"
 	"log"
 	"regexp"
+	"sync"
 	"time"
 
 	"github.com/YakDriver/regexache"
@@ -259,6 +260,12 @@ func resourceTopic() *schema.Resource {
 	}
 }
 
+func constructTopicArn(client *sns.Client, account, region, snsTopicName string) string {
+	return fmt.Sprintf("arn:aws:sns:%s:%s:%s", region, account, snsTopicName)
+}
+
+var snsGlobalMutex sync.Mutex
+
 func resourceTopicCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
 	conn := meta.(*conns.AWSClient).SNSClient(ctx)
@@ -283,6 +290,20 @@ func resourceTopicCreate(ctx context.Context, d *schema.ResourceData, meta inter
 		delete(attributes, topicAttributeNameFIFOTopic)
 	}
 
+	// Look up if the topic already exists
+	snsGlobalMutex.Lock()
+	defer snsGlobalMutex.Unlock()
+	topicArn := constructTopicArn(conn, meta.(*conns.AWSClient).AccountID, meta.(*conns.AWSClient).Region, name)
+
+	_, err = findTopicAttributesWithValidAWSPrincipalsByARN(ctx, conn, topicArn)
+
+	if err == nil {
+		return diag.Errorf("SNS Topic (%s) already exists", name)
+	}
+	if !tfresource.NotFound(err) {
+		return diag.FromErr(err)
+	}
+
 	output, err := conn.CreateTopic(ctx, input)
 
 	// Some partitions (e.g. ISO) may not support tag-on-create.
