From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Fri, 19 Jan 2024 18:08:51 -0500
Subject: [PATCH 38/39] Patch ACM retry to not retry after
 LimitExceededException (#3290)


diff --git a/internal/service/acm/service_package_gen.go b/internal/service/acm/service_package_gen.go
index dbd2989c62..949496c395 100644
--- a/internal/service/acm/service_package_gen.go
+++ b/internal/service/acm/service_package_gen.go
@@ -6,7 +6,9 @@ import (
 	"context"
 
 	aws_sdkv2 "github.com/aws/aws-sdk-go-v2/aws"
+	retry_sdkv2 "github.com/aws/aws-sdk-go-v2/aws/retry"
 	acm_sdkv2 "github.com/aws/aws-sdk-go-v2/service/acm"
+	tfawserr_sdkv2 "github.com/hashicorp/aws-sdk-go-base/v2/tfawserr"
 	"github.com/hashicorp/terraform-provider-aws/internal/conns"
 	"github.com/hashicorp/terraform-provider-aws/internal/types"
 	"github.com/hashicorp/terraform-provider-aws/names"
@@ -60,6 +62,14 @@ func (p *servicePackage) NewClient(ctx context.Context, config map[string]any) (
 		if endpoint := config["endpoint"].(string); endpoint != "" {
 			o.BaseEndpoint = aws_sdkv2.String(endpoint)
 		}
+
+		o.Retryer = conns.AddIsErrorRetryables(cfg.Retryer().(aws_sdkv2.RetryerV2), retry_sdkv2.IsErrorRetryableFunc(func(err error) aws_sdkv2.Ternary {
+			if tfawserr_sdkv2.ErrMessageContains(err, "LimitExceededException", "the maximum number of") &&
+				tfawserr_sdkv2.ErrMessageContains(err, "LimitExceededException", "certificates in the last year") {
+				return aws_sdkv2.FalseTernary
+			}
+			return aws_sdkv2.UnknownTernary // Delegate to configured Retryer.
+		}))
 	}), nil
 }
 
