From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Fri, 9 Feb 2024 14:39:32 -0500
Subject: [PATCH] Patch osis_pipeline tags flags


diff --git a/internal/service/osis/pipeline.go b/internal/service/osis/pipeline.go
index 227eef7004..86f6ebe2f5 100644
--- a/internal/service/osis/pipeline.go
+++ b/internal/service/osis/pipeline.go
@@ -97,7 +97,7 @@ func (r *pipelineResource) Schema(ctx context.Context, request resource.SchemaRe
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"buffer_options": schema.ListNestedBlock{
