From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Tue, 12 Mar 2024 18:05:28 -0400
Subject: [PATCH] securitylake_subscriber tags_all patch


diff --git a/internal/service/securitylake/subscriber.go b/internal/service/securitylake/subscriber.go
index 7390c88b58..4eb91609b5 100644
--- a/internal/service/securitylake/subscriber.go
+++ b/internal/service/securitylake/subscriber.go
@@ -98,7 +98,7 @@ func (r *subscriberResource) Schema(ctx context.Context, request resource.Schema
 			},
 			names.AttrID:      framework.IDAttribute(),
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			names.AttrSource: schema.SetNestedBlock{
