From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Venelin <venelin@pulumi.com>
Date: Mon, 12 Feb 2024 14:02:14 +0000
Subject: [PATCH 42/42] allow lambdas with source_code_hash


diff --git a/internal/service/lambda/function.go b/internal/service/lambda/function.go
index ba93ca7987..b637a47437 100644
--- a/internal/service/lambda/function.go
+++ b/internal/service/lambda/function.go
@@ -169,7 +169,7 @@ func ResourceFunction() *schema.Resource {
 			"filename": {
 				Type:         schema.TypeString,
 				Optional:     true,
-				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket"},
+				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket", "source_code_hash"},
 			},
 			"function_name": {
 				Type:         schema.TypeString,
@@ -208,7 +208,7 @@ func ResourceFunction() *schema.Resource {
 			"image_uri": {
 				Type:         schema.TypeString,
 				Optional:     true,
-				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket"},
+				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket", "source_code_hash"},
 			},
 			"invoke_arn": {
 				Type:     schema.TypeString,
@@ -326,7 +326,7 @@ func ResourceFunction() *schema.Resource {
 			"s3_bucket": {
 				Type:         schema.TypeString,
 				Optional:     true,
-				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket"},
+				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket", "source_code_hash"},
 				RequiredWith: []string{"s3_key"},
 			},
 			"s3_key": {
@@ -375,6 +375,7 @@ func ResourceFunction() *schema.Resource {
 				Optional:         true,
 				Computed:         true,
 				DiffSuppressFunc: verify.SuppressMissingOptionalConfigurationBlock,
+				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket", "source_code_hash"},
 			},
 			"source_code_size": {
 				Type:     schema.TypeInt,
