From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Mon, 1 Apr 2024 17:25:06 -0400
Subject: [PATCH] Patch tags ComputedOnly for m2 resources


diff --git a/internal/service/m2/application.go b/internal/service/m2/application.go
index 221abb4dd5..75ac5b4c94 100644
--- a/internal/service/m2/application.go
+++ b/internal/service/m2/application.go
@@ -105,7 +105,7 @@ func (r *applicationResource) Schema(ctx context.Context, request resource.Schem
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"definition": schema.ListNestedBlock{
diff --git a/internal/service/m2/environment.go b/internal/service/m2/environment.go
index e0ab54314a..c30d3b0386 100644
--- a/internal/service/m2/environment.go
+++ b/internal/service/m2/environment.go
@@ -174,7 +174,7 @@ func (r *environmentResource) Schema(ctx context.Context, request resource.Schem
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"high_availability_config": schema.ListNestedBlock{
