From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: corymhall <43035978+corymhall@users.noreply.github.com>
Date: Wed, 10 Apr 2024 08:49:35 -0400
Subject: [PATCH] restore ECRConn


diff --git a/internal/conns/awsclient_extra.go b/internal/conns/awsclient_extra.go
new file mode 100644
index 0000000000..01a3a5d14d
--- /dev/null
+++ b/internal/conns/awsclient_extra.go
@@ -0,0 +1,13 @@
+package conns
+
+import (
+	"context"
+
+	ecr_sdkv1 "github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+func (c *AWSClient) ECRConn(ctx context.Context) *ecr_sdkv1.ECR {
+	return errs.Must(conn[*ecr_sdkv1.ECR](ctx, c, names.ECR, make(map[string]any)))
+}
