From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: corymhall <43035978+corymhall@users.noreply.github.com>
Date: Thu, 11 Apr 2024 08:06:27 -0400
Subject: [PATCH] update apn info


diff --git a/internal/conns/config.go b/internal/conns/config.go
index 771456c71f..2f807a732e 100644
--- a/internal/conns/config.go
+++ b/internal/conns/config.go
@@ -23,7 +23,6 @@ import (
 	"github.com/hashicorp/terraform-provider-aws/internal/errs/sdkdiag"
 	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
 	"github.com/hashicorp/terraform-provider-aws/names"
-	"github.com/hashicorp/terraform-provider-aws/version"
 )
 
 type Config struct {
@@ -77,10 +76,10 @@ func (c *Config) ConfigureProvider(ctx context.Context, client *AWSClient) (*AWS
 		AccessKey:         c.AccessKey,
 		AllowedAccountIds: c.AllowedAccountIds,
 		APNInfo: &awsbase.APNInfo{
-			PartnerName: "HashiCorp",
+			PartnerName: "Pulumi",
 			Products: []awsbase.UserAgentProduct{
-				{Name: "Terraform", Version: c.TerraformVersion, Comment: "+https://www.terraform.io"},
-				{Name: "terraform-provider-aws", Version: version.ProviderVersion, Comment: "+https://registry.terraform.io/providers/hashicorp/aws"},
+				{Name: "Pulumi", Version: "1.0"},
+				{Name: "Pulumi-Aws", Version: c.TerraformVersion, Comment: "+https://pulumi.com"},
 			},
 		},
 		AssumeRole:                     c.AssumeRole,
