From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Mon, 1 Jul 2024 16:02:48 -0400
Subject: [PATCH] Do not check Computed tags_all


diff --git a/internal/provider/fwprovider/provider.go b/internal/provider/fwprovider/provider.go
index 2d28d90310..b09529ad3e 100644
--- a/internal/provider/fwprovider/provider.go
+++ b/internal/provider/fwprovider/provider.go
@@ -428,12 +428,7 @@ func (p *fwprovider) Resources(ctx context.Context) []func() resource.Resource {
 					errs = append(errs, fmt.Errorf("no `%s` attribute defined in schema: %s", names.AttrTags, typeName))
 					continue
 				}
-				if v, ok := schemaResponse.Schema.Attributes[names.AttrTagsAll]; ok {
-					if v.IsComputed() {
-						errs = append(errs, fmt.Errorf("`%s` attribute must not be Computed: %s", names.AttrTagsAll, typeName))
-						continue
-					}
-				} else {
+				if _, ok := schemaResponse.Schema.Attributes[names.AttrTagsAll]; !ok {
 					errs = append(errs, fmt.Errorf("no `%s` attribute defined in schema: %s", names.AttrTagsAll, typeName))
 					continue
 				}
