From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florian Stadler <florian@pulumi.com>
Date: Tue, 14 May 2024 13:01:51 +0200
Subject: [PATCH] Fix tags_all Computed for aws_datazone_domain


diff --git a/internal/service/datazone/domain.go b/internal/service/datazone/domain.go
index 6b6b258733..843255f209 100644
--- a/internal/service/datazone/domain.go
+++ b/internal/service/datazone/domain.go
@@ -91,7 +91,7 @@ func (r *resourceDomain) Schema(ctx context.Context, req resource.SchemaRequest,
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"single_sign_on": schema.ListNestedBlock{
