From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florian Stadler <florian@pulumi.com>
Date: Tue, 11 Jun 2024 14:02:40 +0200
Subject: [PATCH] Fix tags_all Computed for PF resources


diff --git a/internal/service/paymentcryptography/key.go b/internal/service/paymentcryptography/key.go
index 0c50161ab1..53c35f3c1c 100644
--- a/internal/service/paymentcryptography/key.go
+++ b/internal/service/paymentcryptography/key.go
@@ -113,7 +113,7 @@ func (r *resourceKey) Schema(ctx context.Context, request resource.SchemaRequest
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"key_attributes": schema.SingleNestedBlock{ // nosemgrep:ci.avoid-SingleNestedBlock pre-existing, will be converted
