From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florian Stadler <florian@pulumi.com>
Date: Mon, 24 Jun 2024 12:21:45 +0200
Subject: [PATCH] Fix tags_all Computed for PF resources


diff --git a/internal/service/drs/replication_configuration_template.go b/internal/service/drs/replication_configuration_template.go
index 976dba4ac18..648f2b50030 100644
--- a/internal/service/drs/replication_configuration_template.go
+++ b/internal/service/drs/replication_configuration_template.go
@@ -103,7 +103,7 @@ func (r *replicationConfigurationTemplateResource) Schema(ctx context.Context, r
 
 			"staging_area_tags": tftags.TagsAttributeRequired(),
 			names.AttrTags:      tftags.TagsAttribute(),
-			names.AttrTagsAll:   tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 
 			"use_dedicated_replication_server": schema.BoolAttribute{
 				Required: true,
diff --git a/internal/service/elasticache/reserved_cache_node.go b/internal/service/elasticache/reserved_cache_node.go
index 4aa7fceca63..4ba20a47a74 100644
--- a/internal/service/elasticache/reserved_cache_node.go
+++ b/internal/service/elasticache/reserved_cache_node.go
@@ -111,7 +111,7 @@ func (r *resourceReservedCacheNode) Schema(ctx context.Context, request resource
 				Computed: true,
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"usage_price": schema.Float64Attribute{
 				Computed: true,
 			},
diff --git a/internal/service/pinpointsmsvoicev2/opt_out_list.go b/internal/service/pinpointsmsvoicev2/opt_out_list.go
index 9ed0e93e62d..ba7a696501c 100644
--- a/internal/service/pinpointsmsvoicev2/opt_out_list.go
+++ b/internal/service/pinpointsmsvoicev2/opt_out_list.go
@@ -62,7 +62,7 @@ func (r *optOutListResource) Schema(ctx context.Context, request resource.Schema
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/pinpointsmsvoicev2/phone_number.go b/internal/service/pinpointsmsvoicev2/phone_number.go
index e68971a30d9..68d4fd34476 100644
--- a/internal/service/pinpointsmsvoicev2/phone_number.go
+++ b/internal/service/pinpointsmsvoicev2/phone_number.go
@@ -124,7 +124,7 @@ func (r *phoneNumberResource) Schema(ctx context.Context, request resource.Schem
 				Default:  booldefault.StaticBool(false),
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"two_way_channel_arn": schema.StringAttribute{
 				CustomType: fwtypes.ARNType,
 				Optional:   true,
