From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florian Stadler <florian@pulumi.com>
Date: Mon, 24 Jun 2024 12:21:45 +0200
Subject: [PATCH] Fix tags_all Computed for PF resources


diff --git a/internal/service/drs/replication_configuration_template.go b/internal/service/drs/replication_configuration_template.go
index e23391e0e2..50d387f30b 100644
--- a/internal/service/drs/replication_configuration_template.go
+++ b/internal/service/drs/replication_configuration_template.go
@@ -104,7 +104,7 @@ func (r *replicationConfigurationTemplateResource) Schema(ctx context.Context, r
 
 			"staging_area_tags": tftags.TagsAttributeRequired(),
 			names.AttrTags:      tftags.TagsAttribute(),
-			names.AttrTagsAll:   tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 
 			"use_dedicated_replication_server": schema.BoolAttribute{
 				Required: true,
