From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Mon, 1 Jul 2024 11:21:31 -0400
Subject: [PATCH] Patch tags ComputedOnly for appfabric, networkmonitor,
 networkfirewall


diff --git a/internal/service/appfabric/ingestion.go b/internal/service/appfabric/ingestion.go
index 7ed84d218b..7f092e9590 100644
--- a/internal/service/appfabric/ingestion.go
+++ b/internal/service/appfabric/ingestion.go
@@ -81,7 +81,7 @@ func (r *ingestionResource) Schema(ctx context.Context, request resource.SchemaR
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"tenant_id": schema.StringAttribute{
 				Required: true,
 				Validators: []validator.String{
diff --git a/internal/service/appfabric/ingestion_destination.go b/internal/service/appfabric/ingestion_destination.go
index 08eae51ad4..d0b212ddda 100644
--- a/internal/service/appfabric/ingestion_destination.go
+++ b/internal/service/appfabric/ingestion_destination.go
@@ -85,7 +85,7 @@ func (r *ingestionDestinationResource) Schema(ctx context.Context, request resou
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"destination_configuration": schema.ListNestedBlock{
diff --git a/internal/service/networkfirewall/tls_inspection_configuration.go b/internal/service/networkfirewall/tls_inspection_configuration.go
index a1bd1b65a4..11dc4a67de 100644
--- a/internal/service/networkfirewall/tls_inspection_configuration.go
+++ b/internal/service/networkfirewall/tls_inspection_configuration.go
@@ -118,7 +118,7 @@ func (r *tlsInspectionConfigurationResource) Schema(ctx context.Context, request
 				},
 			},
 			names.AttrTags:                    tftags.TagsAttribute(),
-			names.AttrTagsAll:                 tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"tls_inspection_configuration_id": framework.IDAttribute(),
 			"update_token": schema.StringAttribute{
 				Computed: true,
diff --git a/internal/service/networkmonitor/monitor.go b/internal/service/networkmonitor/monitor.go
index 1cdb74daf6..cf9c98116c 100644
--- a/internal/service/networkmonitor/monitor.go
+++ b/internal/service/networkmonitor/monitor.go
@@ -69,7 +69,7 @@ func (r *monitorResource) Schema(ctx context.Context, request resource.SchemaReq
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/networkmonitor/probe.go b/internal/service/networkmonitor/probe.go
index be5daa5093..701e8c4cbd 100644
--- a/internal/service/networkmonitor/probe.go
+++ b/internal/service/networkmonitor/probe.go
@@ -109,7 +109,7 @@ func (r *probeResource) Schema(ctx context.Context, request resource.SchemaReque
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			names.AttrVPCID: schema.StringAttribute{
 				Computed: true,
 				PlanModifiers: []planmodifier.String{
