From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florian Stadler <florian@pulumi.com>
Date: Fri, 24 May 2024 09:44:07 +0200
Subject: [PATCH] Remove HTTP logger from Lambda resource


diff --git a/internal/service/lambda/service_package_extra.go b/internal/service/lambda/service_package_extra.go
index 54f6aac15a..daa656b829 100644
--- a/internal/service/lambda/service_package_extra.go
+++ b/internal/service/lambda/service_package_extra.go
@@ -6,6 +6,7 @@ import (
 	aws_sdkv2 "github.com/aws/aws-sdk-go-v2/aws"
 	retry_sdkv2 "github.com/aws/aws-sdk-go-v2/aws/retry"
 	lambda_sdkv2 "github.com/aws/aws-sdk-go-v2/service/lambda"
+	"github.com/aws/smithy-go/middleware"
 	tfawserr_sdkv2 "github.com/hashicorp/aws-sdk-go-base/v2/tfawserr"
 	"github.com/hashicorp/terraform-provider-aws/internal/conns"
 	"github.com/hashicorp/terraform-provider-aws/names"
@@ -34,6 +35,10 @@ func (p *servicePackage) NewClient(ctx context.Context, config map[string]any) (
 		if endpoint := config[names.AttrEndpoint].(string); endpoint != "" {
 			o.BaseEndpoint = aws_sdkv2.String(endpoint)
 		}
+		o.APIOptions = append(o.APIOptions, func(stack *middleware.Stack) error {
+			_, err := stack.Deserialize.Remove("TF_AWS_RequestResponseLogger")
+			return err
+		})
 		o.Retryer = conns.AddIsErrorRetryables(cfg.Retryer().(aws_sdkv2.RetryerV2), retry)
 	}), nil
 }
