From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Thu, 18 Jul 2024 13:16:30 -0400
Subject: [PATCH] Fix resourceMatchmakingConfigurationUpdate


diff --git a/internal/service/gamelift/matchmaking_configuration.go b/internal/service/gamelift/matchmaking_configuration.go
index 72f7d31d857..89ecad9db08 100644
--- a/internal/service/gamelift/matchmaking_configuration.go
+++ b/internal/service/gamelift/matchmaking_configuration.go
@@ -309,7 +309,7 @@ func resourceMatchmakingConfigurationUpdate(ctx context.Context, d *schema.Resou
 		input.GameSessionData = aws.String(v.(string))
 	}
 	if v, ok := d.GetOk("game_session_queue_arns"); ok {
-		input.GameSessionQueueArns = expandStringList(v.([]interface{}))
+		input.GameSessionQueueArns = flex.ExpandStringSet(v.(*schema.Set))
 	}
 	if v, ok := d.GetOk("notification_target"); ok {
 		input.NotificationTarget = aws.String(v.(string))
