From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Tue, 23 Jul 2024 17:34:44 -0400
Subject: [PATCH] Patch ComputedOnly for rekognition/stream_processor


diff --git a/internal/service/rekognition/stream_processor.go b/internal/service/rekognition/stream_processor.go
index e7a070f9a4..0e25e4259a 100644
--- a/internal/service/rekognition/stream_processor.go
+++ b/internal/service/rekognition/stream_processor.go
@@ -134,7 +134,7 @@ func (r *resourceStreamProcessor) Schema(ctx context.Context, req resource.Schem
 				DeprecationMessage: "Use 'arn' instead. This attribute will be removed in a future version of the provider.",
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"data_sharing_preference": schema.ListNestedBlock{
