From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Thu, 27 Jun 2024 16:20:44 -0400
Subject: [PATCH] Normalize ecs.TaskDefinition containerDefinitions
 healthCheck.timeout


diff --git a/internal/service/ecs/task_definition_equivalency.go b/internal/service/ecs/task_definition_equivalency.go
index 7aaa1e0679..daf70eec25 100644
--- a/internal/service/ecs/task_definition_equivalency.go
+++ b/internal/service/ecs/task_definition_equivalency.go
@@ -50,6 +50,10 @@ func (cd containerDefinitions) reduce(isAWSVPC bool) {
 		if def.Essential == nil {
 			cd[i].Essential = aws.Bool(true)
 		}
+		if def.HealthCheck != nil && def.HealthCheck.Timeout == nil {
+			five := int32(5)
+			def.HealthCheck.Timeout = &five
+		}
 		for j, pm := range def.PortMappings {
 			if pm.Protocol == awstypes.TransportProtocolTcp {
 				cd[i].PortMappings[j].Protocol = ""
