From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Tue, 23 Jul 2024 17:34:44 -0400
Subject: [PATCH] Patch ComputedOnly for rekognition/stream_processor


diff --git a/internal/service/rekognition/stream_processor.go b/internal/service/rekognition/stream_processor.go
index 77321677cc..40e2552d3b 100644
--- a/internal/service/rekognition/stream_processor.go
+++ b/internal/service/rekognition/stream_processor.go
@@ -136,7 +136,7 @@ func (r *resourceStreamProcessor) Schema(ctx context.Context, req resource.Schem
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"data_sharing_preference": schema.ListNestedBlock{
