From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Mon, 1 Jul 2024 11:21:31 -0400
Subject: [PATCH] Patch tags ComputedOnly for appfabric, networkmonitor,
 networkfirewall


diff --git a/internal/service/appfabric/ingestion.go b/internal/service/appfabric/ingestion.go
index b3d22c2512..b40e465917 100644
--- a/internal/service/appfabric/ingestion.go
+++ b/internal/service/appfabric/ingestion.go
@@ -77,7 +77,7 @@ func (r *ingestionResource) Schema(ctx context.Context, request resource.SchemaR
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"tenant_id": schema.StringAttribute{
 				Required: true,
 				Validators: []validator.String{
diff --git a/internal/service/appfabric/ingestion_destination.go b/internal/service/appfabric/ingestion_destination.go
index a2b6c366c8..31dd84e1eb 100644
--- a/internal/service/appfabric/ingestion_destination.go
+++ b/internal/service/appfabric/ingestion_destination.go
@@ -81,7 +81,7 @@ func (r *ingestionDestinationResource) Schema(ctx context.Context, request resou
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"destination_configuration": schema.ListNestedBlock{
diff --git a/internal/service/networkfirewall/tls_inspection_configuration.go b/internal/service/networkfirewall/tls_inspection_configuration.go
index d4444801b5..a2ae1ed59a 100644
--- a/internal/service/networkfirewall/tls_inspection_configuration.go
+++ b/internal/service/networkfirewall/tls_inspection_configuration.go
@@ -118,7 +118,7 @@ func (r *tlsInspectionConfigurationResource) Schema(ctx context.Context, request
 				},
 			},
 			names.AttrTags:                    tftags.TagsAttribute(),
-			names.AttrTagsAll:                 tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"tls_inspection_configuration_id": framework.IDAttribute(),
 			"update_token": schema.StringAttribute{
 				Computed: true,
diff --git a/internal/service/networkmonitor/monitor.go b/internal/service/networkmonitor/monitor.go
index c47a783686..a940717da7 100644
--- a/internal/service/networkmonitor/monitor.go
+++ b/internal/service/networkmonitor/monitor.go
@@ -69,7 +69,7 @@ func (r *monitorResource) Schema(ctx context.Context, request resource.SchemaReq
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/networkmonitor/probe.go b/internal/service/networkmonitor/probe.go
index 7d11f83838..47d5b89254 100644
--- a/internal/service/networkmonitor/probe.go
+++ b/internal/service/networkmonitor/probe.go
@@ -109,7 +109,7 @@ func (r *probeResource) Schema(ctx context.Context, request resource.SchemaReque
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			names.AttrVPCID: schema.StringAttribute{
 				Computed: true,
 				PlanModifiers: []planmodifier.String{
