From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florian Stadler <florian@pulumi.com>
Date: Mon, 12 Aug 2024 11:11:50 +0200
Subject: [PATCH] Patch tags_all computed for aws_rds_integration


diff --git a/internal/service/rds/integration.go b/internal/service/rds/integration.go
index e83c54079f..07ad3e55b1 100644
--- a/internal/service/rds/integration.go
+++ b/internal/service/rds/integration.go
@@ -104,7 +104,7 @@ func (r *integrationResource) Schema(ctx context.Context, request resource.Schem
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			names.AttrTargetARN: schema.StringAttribute{
 				CustomType: fwtypes.ARNType,
 				Required:   true,
