From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Mon, 1 Jul 2024 14:54:16 -0400
Subject: [PATCH] Transform PF provider to make tags not computed


diff --git a/shim/shim.go b/shim/shim.go
index b94f722d26..512c20315b 100644
--- a/shim/shim.go
+++ b/shim/shim.go
@@ -25,7 +25,7 @@ func NewUpstreamProvider(ctx context.Context) (UpstreamProvider, error) {
 	if primary != nil {
 		markTagsAllNotComputedForResources(primary)
 	}
-	pf := fwprovider.New(primary)
+	pf := &transformedFrameworkProvider{fwprovider.New(primary)}
 	return UpstreamProvider{
 		SDKV2Provider:           primary,
 		PluginFrameworkProvider: pf,
