From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: corymhall <43035978+corymhall@users.noreply.github.com>
Date: Fri, 2 Aug 2024 09:13:02 -0400
Subject: [PATCH] Patch tags_all Computed for chatbot, timestreaminfluxdb


diff --git a/internal/service/chatbot/slack_channel_configuration.go b/internal/service/chatbot/slack_channel_configuration.go
index edd34ca88f..97e8afe082 100644
--- a/internal/service/chatbot/slack_channel_configuration.go
+++ b/internal/service/chatbot/slack_channel_configuration.go
@@ -104,7 +104,7 @@ func (r *slackChannelConfigurationResource) Schema(ctx context.Context, request
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"user_authorization_required": schema.BoolAttribute{
 				Optional: true,
 				Computed: true,
diff --git a/internal/service/chatbot/teams_channel_configuration.go b/internal/service/chatbot/teams_channel_configuration.go
index 8fe51bfe4e..77887fcec9 100644
--- a/internal/service/chatbot/teams_channel_configuration.go
+++ b/internal/service/chatbot/teams_channel_configuration.go
@@ -101,7 +101,7 @@ func (r *teamsChannelConfigurationResource) Schema(ctx context.Context, request
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"team_id": schema.StringAttribute{
 				Required: true,
 			},
diff --git a/internal/service/timestreaminfluxdb/db_instance.go b/internal/service/timestreaminfluxdb/db_instance.go
index 60158012a7..7ce0339236 100644
--- a/internal/service/timestreaminfluxdb/db_instance.go
+++ b/internal/service/timestreaminfluxdb/db_instance.go
@@ -188,7 +188,7 @@ func (r *resourceDBInstance) Schema(ctx context.Context, req resource.SchemaRequ
 					and per region.`,
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"organization": schema.StringAttribute{
 				Required: true,
 				PlanModifiers: []planmodifier.String{
