From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florian Stadler <florian@pulumi.com>
Date: Tue, 20 Aug 2024 14:14:42 +0200
Subject: [PATCH] Patch Tags for bedrock and pinpoint


diff --git a/internal/service/bedrock/guardrail.go b/internal/service/bedrock/guardrail.go
index 7d83ca812a..a9abb0d2ee 100644
--- a/internal/service/bedrock/guardrail.go
+++ b/internal/service/bedrock/guardrail.go
@@ -122,7 +122,7 @@ func (r *resourceGuardrail) Schema(ctx context.Context, req resource.SchemaReque
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			names.AttrVersion: schema.StringAttribute{
 				Computed: true,
 				PlanModifiers: []planmodifier.String{
diff --git a/internal/service/pinpoint/email_template.go b/internal/service/pinpoint/email_template.go
index e20b0d1c10..dd70f5a417 100644
--- a/internal/service/pinpoint/email_template.go
+++ b/internal/service/pinpoint/email_template.go
@@ -63,7 +63,7 @@ func (r *resourceEmailTemplate) Schema(ctx context.Context, req resource.SchemaR
 				Required: true,
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"email_template": schema.ListNestedBlock{
