From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Mon, 14 Oct 2024 15:15:47 -0400
Subject: [PATCH] Patch ComputedOnly for route53profiles and more


diff --git a/internal/service/backup/restore_testing_plan.go b/internal/service/backup/restore_testing_plan.go
index e19d23122a..550c9fa3da 100644
--- a/internal/service/backup/restore_testing_plan.go
+++ b/internal/service/backup/restore_testing_plan.go
@@ -85,7 +85,7 @@ func (r *restoreTestingPlanResource) Schema(ctx context.Context, request resourc
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"recovery_point_selection": schema.ListNestedBlock{
diff --git a/internal/service/pinpointsmsvoicev2/configuration_set.go b/internal/service/pinpointsmsvoicev2/configuration_set.go
index b1bd5bcd79..2a7e5669b7 100644
--- a/internal/service/pinpointsmsvoicev2/configuration_set.go
+++ b/internal/service/pinpointsmsvoicev2/configuration_set.go
@@ -71,7 +71,7 @@ func (r *configurationSetResource) Schema(ctx context.Context, request resource.
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/route53profiles/association.go b/internal/service/route53profiles/association.go
index 77ad722ee8..b827a6ba7f 100644
--- a/internal/service/route53profiles/association.go
+++ b/internal/service/route53profiles/association.go
@@ -109,7 +109,7 @@ func (r *resourceAssociation) Schema(ctx context.Context, req resource.SchemaReq
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			names.AttrTimeouts: timeouts.Block(ctx, timeouts.Opts{
diff --git a/internal/service/route53profiles/profile.go b/internal/service/route53profiles/profile.go
index 6c0b3d3e0a..3f90c6dc65 100644
--- a/internal/service/route53profiles/profile.go
+++ b/internal/service/route53profiles/profile.go
@@ -90,7 +90,7 @@ func (r *resourceProfile) Schema(ctx context.Context, req resource.SchemaRequest
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			names.AttrTimeouts: timeouts.Block(ctx, timeouts.Opts{
