From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Mon, 14 Oct 2024 15:15:47 -0400
Subject: [PATCH] Patch ComputedOnly for route53profiles and more


diff --git a/internal/service/backup/restore_testing_plan.go b/internal/service/backup/restore_testing_plan.go
index 7d9155b5f0..c93f05adc5 100644
--- a/internal/service/backup/restore_testing_plan.go
+++ b/internal/service/backup/restore_testing_plan.go
@@ -89,7 +89,7 @@ func (r *restoreTestingPlanResource) Schema(ctx context.Context, request resourc
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"recovery_point_selection": schema.ListNestedBlock{
diff --git a/internal/service/pinpointsmsvoicev2/configuration_set.go b/internal/service/pinpointsmsvoicev2/configuration_set.go
index b4bd9f9d39..61334ec528 100644
--- a/internal/service/pinpointsmsvoicev2/configuration_set.go
+++ b/internal/service/pinpointsmsvoicev2/configuration_set.go
@@ -75,7 +75,7 @@ func (r *configurationSetResource) Schema(ctx context.Context, request resource.
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/route53profiles/association.go b/internal/service/route53profiles/association.go
index 7511a8a554..3a72dd67f3 100644
--- a/internal/service/route53profiles/association.go
+++ b/internal/service/route53profiles/association.go
@@ -114,7 +114,7 @@ func (r *resourceAssociation) Schema(ctx context.Context, req resource.SchemaReq
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			names.AttrTimeouts: timeouts.Block(ctx, timeouts.Opts{
diff --git a/internal/service/route53profiles/profile.go b/internal/service/route53profiles/profile.go
index b65485fd1b..529f1589f8 100644
--- a/internal/service/route53profiles/profile.go
+++ b/internal/service/route53profiles/profile.go
@@ -94,7 +94,7 @@ func (r *resourceProfile) Schema(ctx context.Context, req resource.SchemaRequest
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			names.AttrTimeouts: timeouts.Block(ctx, timeouts.Opts{
