From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Fri, 25 Oct 2024 15:00:58 -0400
Subject: [PATCH] Temporarily remove iam_role managed_policy_arn deprecation


diff --git a/internal/service/iam/role.go b/internal/service/iam/role.go
index a2b746356e..e6171b295a 100644
--- a/internal/service/iam/role.go
+++ b/internal/service/iam/role.go
@@ -131,11 +131,6 @@ func resourceRole() *schema.Resource {
 				Type:     schema.TypeSet,
 				Optional: true,
 				Computed: true,
-				Deprecated: "The managed_policy_arns argument is deprecated. " +
-					"Use the aws_iam_role_policy_attachment resource instead. If Terraform should" +
-					"exclusively manage all managed policy attachments (the current " +
-					"behavior of this argument), use the aws_iam_role_policy_attachments_exclusive " +
-					"resource as well.",
 				Elem: &schema.Schema{
 					Type:         schema.TypeString,
 					ValidateFunc: verify.ValidARN,
