From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Mon, 28 Oct 2024 17:49:18 -0400
Subject: [PATCH] Restore AWS Go SDK v1 session


diff --git a/internal/conns/awsclient.go b/internal/conns/awsclient.go
index 932e5dff771..ff1bdc73d15 100644
--- a/internal/conns/awsclient.go
+++ b/internal/conns/awsclient.go
@@ -321,6 +321,7 @@ func (c *AWSClient) apiClientConfig(ctx context.Context, servicePackageName stri
 		"aws_sdkv2_config": c.awsConfig,
 		"endpoint":         c.endpoints[servicePackageName],
 		"partition":        c.Partition(ctx),
+		"session":          c.session,
 	}
 	switch servicePackageName {
 	case names.S3:
