From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Mon, 4 Nov 2024 14:45:56 -0500
Subject: [PATCH] Patch out PF tags for ssmquicksetup


diff --git a/internal/service/ssmquicksetup/configuration_manager.go b/internal/service/ssmquicksetup/configuration_manager.go
index 6cc0a58961..f116979abf 100644
--- a/internal/service/ssmquicksetup/configuration_manager.go
+++ b/internal/service/ssmquicksetup/configuration_manager.go
@@ -75,7 +75,7 @@ func (r *resourceConfigurationManager) Schema(ctx context.Context, req resource.
 				ElementType: fwtypes.NewObjectTypeOf[statusSummaryModel](ctx),
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"configuration_definition": schema.ListNestedBlock{
