From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Mon, 28 Oct 2024 17:49:18 -0400
Subject: [PATCH] Restore AWS Go SDK v1 session


diff --git a/internal/conns/awsclient.go b/internal/conns/awsclient.go
index ddfe9d8849..6e2fc6ff46 100644
--- a/internal/conns/awsclient.go
+++ b/internal/conns/awsclient.go
@@ -253,6 +253,7 @@ func (c *AWSClient) apiClientConfig(_ context.Context, servicePackageName string
 		"aws_sdkv2_config": c.awsConfig,
 		"endpoint":         c.endpoints[servicePackageName],
 		"partition":        c.Partition,
+		"session":          c.session,
 	}
 	switch servicePackageName {
 	case names.S3:
