From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florian Stadler <florian@pulumi.com>
Date: Thu, 5 Dec 2024 20:46:24 +0100
Subject: [PATCH] Fix tags_all Computed for CodeConnections Host and Connection


diff --git a/internal/service/codeconnections/connection.go b/internal/service/codeconnections/connection.go
index d232de6a8d..68228b216d 100644
--- a/internal/service/codeconnections/connection.go
+++ b/internal/service/codeconnections/connection.go
@@ -107,7 +107,7 @@ func (r *connectionResource) Schema(ctx context.Context, req resource.SchemaRequ
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			names.AttrTimeouts: timeouts.Block(ctx, timeouts.Opts{
diff --git a/internal/service/codeconnections/host.go b/internal/service/codeconnections/host.go
index 4a86607bdc..15581c4985 100644
--- a/internal/service/codeconnections/host.go
+++ b/internal/service/codeconnections/host.go
@@ -86,7 +86,7 @@ func (r *hostResource) Schema(ctx context.Context, req resource.SchemaRequest, r
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			names.AttrVPCConfiguration: schema.ListNestedBlock{
