From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florian Stadler <florian@pulumi.com>
Date: Fri, 13 Dec 2024 10:03:47 +0100
Subject: [PATCH] Fix tags_all Computed for PF resources


diff --git a/internal/service/bedrock/inference_profile.go b/internal/service/bedrock/inference_profile.go
index f797ba7013..fee74d9e3c 100644
--- a/internal/service/bedrock/inference_profile.go
+++ b/internal/service/bedrock/inference_profile.go
@@ -115,7 +115,7 @@ func (r *resourceInferenceProfile) Schema(ctx context.Context, req resource.Sche
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"model_source": schema.ListNestedBlock{
diff --git a/internal/service/logs/anomaly_detector.go b/internal/service/logs/anomaly_detector.go
index d69e4e44c8..d1a6eae711 100644
--- a/internal/service/logs/anomaly_detector.go
+++ b/internal/service/logs/anomaly_detector.go
@@ -85,7 +85,7 @@ func (r *anomalyDetectorResource) Schema(ctx context.Context, request resource.S
 				ElementType: types.StringType,
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/servicecatalogappregistry/application.go b/internal/service/servicecatalogappregistry/application.go
index 56e6db789f..058f43711b 100644
--- a/internal/service/servicecatalogappregistry/application.go
+++ b/internal/service/servicecatalogappregistry/application.go
@@ -66,7 +66,7 @@ func (r *resourceApplication) Schema(ctx context.Context, req resource.SchemaReq
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/servicecatalogappregistry/attribute_group.go b/internal/service/servicecatalogappregistry/attribute_group.go
index a89b455ba3..5c86908c7c 100644
--- a/internal/service/servicecatalogappregistry/attribute_group.go
+++ b/internal/service/servicecatalogappregistry/attribute_group.go
@@ -72,7 +72,7 @@ func (r *resourceAttributeGroup) Schema(ctx context.Context, req resource.Schema
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
