From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florian Stadler <florian@pulumi.com>
Date: Fri, 13 Dec 2024 10:03:47 +0100
Subject: [PATCH] Fix tags_all Computed for PF resources


diff --git a/internal/service/bedrock/inference_profile.go b/internal/service/bedrock/inference_profile.go
index eb2348bc68..305435e1b7 100644
--- a/internal/service/bedrock/inference_profile.go
+++ b/internal/service/bedrock/inference_profile.go
@@ -119,7 +119,7 @@ func (r *resourceInferenceProfile) Schema(ctx context.Context, req resource.Sche
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"model_source": schema.ListNestedBlock{
diff --git a/internal/service/logs/anomaly_detector.go b/internal/service/logs/anomaly_detector.go
index 788672e044..e637e203c3 100644
--- a/internal/service/logs/anomaly_detector.go
+++ b/internal/service/logs/anomaly_detector.go
@@ -95,7 +95,7 @@ func (r *resourceAnomalyDetector) Schema(ctx context.Context, req resource.Schem
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/servicecatalogappregistry/application.go b/internal/service/servicecatalogappregistry/application.go
index 2844029f5c..735bfe81bb 100644
--- a/internal/service/servicecatalogappregistry/application.go
+++ b/internal/service/servicecatalogappregistry/application.go
@@ -70,7 +70,7 @@ func (r *resourceApplication) Schema(ctx context.Context, req resource.SchemaReq
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/servicecatalogappregistry/attribute_group.go b/internal/service/servicecatalogappregistry/attribute_group.go
index e1719b8cfa..4e28fbd51a 100644
--- a/internal/service/servicecatalogappregistry/attribute_group.go
+++ b/internal/service/servicecatalogappregistry/attribute_group.go
@@ -76,7 +76,7 @@ func (r *resourceAttributeGroup) Schema(ctx context.Context, req resource.Schema
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
