From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florian Stadler <florian@pulumi.com>
Date: Fri, 20 Dec 2024 09:36:29 +0100
Subject: [PATCH] Fix tags_all Computed for PF resources


diff --git a/internal/service/apigateway/domain_name_access_association.go b/internal/service/apigateway/domain_name_access_association.go
index c1945339731..6b6c8681583 100644
--- a/internal/service/apigateway/domain_name_access_association.go
+++ b/internal/service/apigateway/domain_name_access_association.go
@@ -74,7 +74,7 @@ func (r *domainNameAccessAssociationResource) Schema(ctx context.Context, reques
 			},
 			names.AttrID:      framework.IDAttribute(),
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/cloudfront/vpc_origin.go b/internal/service/cloudfront/vpc_origin.go
index c5ccae5b51a..b3eb73bb756 100644
--- a/internal/service/cloudfront/vpc_origin.go
+++ b/internal/service/cloudfront/vpc_origin.go
@@ -60,7 +60,7 @@ func (r *vpcOriginResource) Schema(ctx context.Context, request resource.SchemaR
 				Computed: true,
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"vpc_origin_endpoint_config": schema.ListNestedBlock{
diff --git a/internal/service/memorydb/multi_region_cluster.go b/internal/service/memorydb/multi_region_cluster.go
index 42b0fc1d42e..e3b5b68ca07 100644
--- a/internal/service/memorydb/multi_region_cluster.go
+++ b/internal/service/memorydb/multi_region_cluster.go
@@ -126,7 +126,7 @@ func (r *multiRegionClusterResource) Schema(ctx context.Context, request resourc
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"tls_enabled": schema.BoolAttribute{
 				Optional: true,
 				Computed: true,
diff --git a/internal/service/networkmanager/dx_gateway_attachment.go b/internal/service/networkmanager/dx_gateway_attachment.go
index 508ce646c35..23637d8c5eb 100644
--- a/internal/service/networkmanager/dx_gateway_attachment.go
+++ b/internal/service/networkmanager/dx_gateway_attachment.go
@@ -111,7 +111,7 @@ func (r *directConnectGatewayAttachmentResource) Schema(ctx context.Context, req
 				Computed: true,
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			names.AttrTimeouts: timeouts.Block(ctx, timeouts.Opts{
diff --git a/internal/service/rds/cluster_snapshot_copy.go b/internal/service/rds/cluster_snapshot_copy.go
index f9044db1ed7..db79f0c198d 100644
--- a/internal/service/rds/cluster_snapshot_copy.go
+++ b/internal/service/rds/cluster_snapshot_copy.go
@@ -146,7 +146,7 @@ func (r *resourceClusterSnapshotCopy) Schema(ctx context.Context, req resource.S
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"target_db_cluster_snapshot_identifier": schema.StringAttribute{
 				Required: true,
 				PlanModifiers: []planmodifier.String{
