From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anton Tayanovskyy <anton@pulumi.com>
Date: Fri, 10 Jan 2025 14:24:40 -0500
Subject: [PATCH] Patch tags for logs, cleanrooms, vpclattice


diff --git a/internal/service/cleanrooms/membership.go b/internal/service/cleanrooms/membership.go
index ee4f03039e..991000a637 100644
--- a/internal/service/cleanrooms/membership.go
+++ b/internal/service/cleanrooms/membership.go
@@ -111,7 +111,7 @@ func (r *resourceMembership) Schema(ctx context.Context, _ resource.SchemaReques
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"update_time": schema.StringAttribute{
 				CustomType: timetypes.RFC3339Type{},
 				Computed:   true,
diff --git a/internal/service/logs/delivery.go b/internal/service/logs/delivery.go
index d749138f74..29e6891331 100644
--- a/internal/service/logs/delivery.go
+++ b/internal/service/logs/delivery.go
@@ -93,7 +93,7 @@ func (r *deliveryResource) Schema(ctx context.Context, request resource.SchemaRe
 			},
 			"s3_delivery_configuration": framework.ResourceOptionalComputedListOfObjectsAttribute[s3DeliveryConfigurationModel](ctx, 1, s3DeliveryConfigurationListOptions, listplanmodifier.UseStateForUnknown()),
 			names.AttrTags:              tftags.TagsAttribute(),
-			names.AttrTagsAll:           tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll:           tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/logs/delivery_destination.go b/internal/service/logs/delivery_destination.go
index 5eb2035058..e48697cc2f 100644
--- a/internal/service/logs/delivery_destination.go
+++ b/internal/service/logs/delivery_destination.go
@@ -69,7 +69,7 @@ func (r *deliveryDestinationResource) Schema(ctx context.Context, request resour
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"delivery_destination_configuration": schema.ListNestedBlock{
diff --git a/internal/service/logs/delivery_source.go b/internal/service/logs/delivery_source.go
index 3b172ef95a..16b9173d00 100644
--- a/internal/service/logs/delivery_source.go
+++ b/internal/service/logs/delivery_source.go
@@ -83,7 +83,7 @@ func (r *deliverySourceResource) Schema(ctx context.Context, request resource.Sc
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/vpclattice/resource_gateway.go b/internal/service/vpclattice/resource_gateway.go
index f5c0174c72..38f775fb8f 100644
--- a/internal/service/vpclattice/resource_gateway.go
+++ b/internal/service/vpclattice/resource_gateway.go
@@ -98,7 +98,7 @@ func (r *resourceGatewayResource) Schema(ctx context.Context, request resource.S
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			names.AttrVPCID: schema.StringAttribute{
 				Required: true,
 				PlanModifiers: []planmodifier.String{
