From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: corymhall <43035978+corymhall@users.noreply.github.com>
Date: Tue, 21 Jan 2025 10:53:09 -0500
Subject: [PATCH] Patch tags for mediapackagev2 channel_group


diff --git a/internal/service/mediapackagev2/channel_group.go b/internal/service/mediapackagev2/channel_group.go
index ef6ee09042..166ad87546 100644
--- a/internal/service/mediapackagev2/channel_group.go
+++ b/internal/service/mediapackagev2/channel_group.go
@@ -70,7 +70,7 @@ func (r *resourceChannelGroup) Schema(ctx context.Context, request resource.Sche
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 
