From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: guineveresaenger <guinevere@pulumi.com>
Date: Fri, 21 Feb 2025 10:44:05 -0800
Subject: [PATCH] Remove Computed Only for Cloudwatch and Qbusiness


diff --git a/internal/service/cloudwatch/contributor_insight_rule.go b/internal/service/cloudwatch/contributor_insight_rule.go
index 8e077de1e9..a4318698d4 100644
--- a/internal/service/cloudwatch/contributor_insight_rule.go
+++ b/internal/service/cloudwatch/contributor_insight_rule.go
@@ -62,7 +62,7 @@ func (r *resourceContributorInsightRule) Schema(ctx context.Context, req resourc
 				CustomType: fwtypes.StringEnumType[stateValue](),
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/cloudwatch/contributor_managed_insight_rule.go b/internal/service/cloudwatch/contributor_managed_insight_rule.go
index b4e1a4de14..ff346c07ec 100644
--- a/internal/service/cloudwatch/contributor_managed_insight_rule.go
+++ b/internal/service/cloudwatch/contributor_managed_insight_rule.go
@@ -85,7 +85,7 @@ func (r *resourceContributorManagedInsightRule) Schema(ctx context.Context, req
 				Computed: true,
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 	}
 }
diff --git a/internal/service/qbusiness/application.go b/internal/service/qbusiness/application.go
index 0ddf636ff1..48ded15cc3 100644
--- a/internal/service/qbusiness/application.go
+++ b/internal/service/qbusiness/application.go
@@ -89,7 +89,7 @@ func (r *resourceApplication) Schema(ctx context.Context, req resource.SchemaReq
 				Required:    true,
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			"attachments_configuration": schema.ListNestedBlock{
