From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Venelin <venelin@pulumi.com>
Date: Tue, 4 Mar 2025 20:27:32 +0200
Subject: [PATCH] patch tags for nepture graph and rds shard group


diff --git a/internal/service/neptunegraph/graph.go b/internal/service/neptunegraph/graph.go
index b68cc661f6..b2885f3616 100644
--- a/internal/service/neptunegraph/graph.go
+++ b/internal/service/neptunegraph/graph.go
@@ -144,7 +144,7 @@ func (r *graphResource) Schema(ctx context.Context, request resource.SchemaReque
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			names.AttrTimeouts: timeouts.Block(ctx, timeouts.Opts{
diff --git a/internal/service/rds/shard_group.go b/internal/service/rds/shard_group.go
index ab15481439..5326cc21cd 100644
--- a/internal/service/rds/shard_group.go
+++ b/internal/service/rds/shard_group.go
@@ -113,7 +113,7 @@ func (r *shardGroupResource) Schema(ctx context.Context, request resource.Schema
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 		},
 		Blocks: map[string]schema.Block{
 			names.AttrTimeouts: timeouts.Block(ctx, timeouts.Opts{
