From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: guineveresaenger <guinevere@pulumi.com>
Date: Mon, 24 Mar 2025 16:11:19 -0700
Subject: [PATCH] Fix tags_all for Athena Capacity Reservation


diff --git a/internal/service/athena/capacity_reservation.go b/internal/service/athena/capacity_reservation.go
index 8d4fe30cb1..d327ce048c 100644
--- a/internal/service/athena/capacity_reservation.go
+++ b/internal/service/athena/capacity_reservation.go
@@ -69,7 +69,7 @@ func (r *resourceCapacityReservation) Schema(ctx context.Context, req resource.S
 				Computed: true,
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			"target_dpus": schema.Int32Attribute{
 				Required: true,
 				Validators: []validator.Int32{
