From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Venelin <venelin@pulumi.com>
Date: Fri, 7 Mar 2025 18:28:52 +0200
Subject: [PATCH] disable plan modification for tags in PF resources


diff --git a/internal/provider/fwprovider/wrap.go b/internal/provider/fwprovider/wrap.go
index fc9a9c859c..a95462a70f 100644
--- a/internal/provider/fwprovider/wrap.go
+++ b/internal/provider/fwprovider/wrap.go
@@ -264,7 +264,7 @@ func (w *wrappedResource) ImportState(ctx context.Context, request resource.Impo
 func (w *wrappedResource) ModifyPlan(ctx context.Context, request resource.ModifyPlanRequest, response *resource.ModifyPlanResponse) {
 	ctx = w.opts.bootstrapContext(ctx, w.meta)
 
-	if w.opts.usesTransparentTagging {
+	if w.opts.usesTransparentTagging && false {
 		w.setTagsAll(ctx, request, response)
 		if response.Diagnostics.HasError() {
 			return
