From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Venelin <venelin@pulumi.com>
Date: Tue, 8 Apr 2025 14:18:40 +0300
Subject: [PATCH] patch gamelift tags back


diff --git a/internal/service/gamelift/tags_gen.go b/internal/service/gamelift/tags_gen.go
index 19ffbdbd57..1640243252 100644
--- a/internal/service/gamelift/tags_gen.go
+++ b/internal/service/gamelift/tags_gen.go
@@ -144,3 +144,19 @@ func updateTags(ctx context.Context, conn *gamelift.Client, identifier string, o
 func (p *servicePackage) UpdateTags(ctx context.Context, meta any, identifier string, oldTags, newTags any) error {
 	return updateTags(ctx, meta.(*conns.AWSClient).GameLiftClient(ctx), identifier, oldTags, newTags)
 }
+
+// Tags returns gamelift service tags.
+func Tags(tags tftags.KeyValueTags) []awstypes.Tag {
+	result := make([]awstypes.Tag, 0, len(tags))
+
+	for k, v := range tags.Map() {
+		tag := awstypes.Tag{
+			Key:   aws.String(k),
+			Value: aws.String(v),
+		}
+
+		result = append(result, tag)
+	}
+
+	return result
+}
\ No newline at end of file
