From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Zaid Ajaj <zaid.naom@gmail.com>
Date: Wed, 23 Apr 2025 19:08:58 +0200
Subject: [PATCH] Fix tags_all Computed for PF resources


diff --git a/internal/service/redshift/integration.go b/internal/service/redshift/integration.go
index 691478c7af..e0745b2847 100644
--- a/internal/service/redshift/integration.go
+++ b/internal/service/redshift/integration.go
@@ -81,7 +81,7 @@ func (r *integrationResource) Schema(ctx context.Context, req resource.SchemaReq
 				},
 			},
 			names.AttrTags:    tftags.TagsAttribute(),
-			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			names.AttrTagsAll: tftags.TagsAttribute(),
 			names.AttrTargetARN: schema.StringAttribute{
 				CustomType: fwtypes.ARNType,
 				Required:   true,
