From d6e0d16dd1dafe254e58973843d4a28bb84afffd Mon Sep 17 00:00:00 2001
From: Daniel Bradley <daniel@pulumi.com>
Date: Fri, 4 Nov 2022 17:06:11 +0000
Subject: [PATCH 05/29] De-deprecate bucket_object

---
 internal/service/s3/bucket_object.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/internal/service/s3/bucket_object.go b/internal/service/s3/bucket_object.go
index 1340e045ea..f0a1ce7041 100644
--- a/internal/service/s3/bucket_object.go
+++ b/internal/service/s3/bucket_object.go
@@ -60,7 +60,7 @@ func ResourceBucketObject() *schema.Resource {
 				ValidateFunc: validation.StringInSlice(s3.ObjectCannedACL_Values(), false),
 			},
 			"bucket": {
-				Deprecated:   "Use the aws_s3_object resource instead",
+				// FORK: Stack72 removed the deprecation warning as it was misleading for Pulumi users
 				Type:         schema.TypeString,
 				Required:     true,
 				ForceNew:     true,
@@ -117,7 +117,7 @@ func ResourceBucketObject() *schema.Resource {
 				Default:  false,
 			},
 			"key": {
-				Deprecated:   "Use the aws_s3_object resource instead",
+				// FORK: Stack72 removed the deprecation warning as it was misleading for Pulumi users
 				Type:         schema.TypeString,
 				Required:     true,
 				ForceNew:     true,
