From 86d7a60259026b352eecc02dbb739b4cc3ed2e58 Mon Sep 17 00:00:00 2001
From: Richard Shade <richard@pulumi.com>
Date: Fri, 6 Jan 2023 15:31:22 -0600
Subject: [PATCH 19/29] fix: Adding back in removeAddons

---
 internal/service/eks/cluster.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/internal/service/eks/cluster.go b/internal/service/eks/cluster.go
index 06d996e1ef..bd2a3c8b9d 100644
--- a/internal/service/eks/cluster.go
+++ b/internal/service/eks/cluster.go
@@ -378,6 +378,10 @@ func resourceClusterCreate(ctx context.Context, d *schema.ResourceData, meta int
 		return diag.Errorf("waiting for EKS Cluster (%s) create: %s", d.Id(), err)
 	}
 
+	if err = removeAddons(d, conn); err != nil {
+		return diag.Errorf("removing addons (%s): %s", d.Id(), err)
+	}
+
 	return resourceClusterRead(ctx, d, meta)
 }
 
